---
title: "Jin Project 2"
subtitle: "Elevated CO2 and phytate"
output: 
  bookdown::html_document2:
    fig_caption: yes
    toc: yes
    toc_float: true
    number_sections: false
always_allow_html: true
editor_options: 
  chunk_output_type: inline
knit: (function(input_file, encoding) {
  out_dir <- 'docs';
  rmarkdown::render(input_file,
  encoding=encoding,
  output_file=file.path(dirname(input_file), out_dir, 'index.html'))})
  
  
---

```{r packages, message=FALSE, warning=FALSE, echo=FALSE}
#mywd
#knit: (function(input_file, encoding) {
#  out_dir <- 'docs';
#  rmarkdown::render(input_file,
# encoding=encoding,
# output_file=file.path(dirname(input_file), out_dir, 'index.html'))})
#https://resources.github.com/whitepapers/github-and-rstudio/ #info on github page creation
setwd("~/Documents/Study/LaTrobe/Research/phD/Jin_Project2_Sand/Jincollaboration")
#mypackages
library(ggrepel)
library(qiime2R)
library(tidyverse)
library(stringr)
library(ggplot2)
theme_set(theme_bw())  
library(vegan)
library(ggpubr)
library(RColorBrewer)
library(phyloseq)
library(psych)
library(reshape2)
library(ggfortify)
library(factoextra)
library(FactoMineR)
library(maigesPack)
library(knitr)
library(microbiome)
library(SpiecEasi)
library(bookdown)
library(tinytex)
library(jakR)
library(kableExtra)
library(DESeq2)
options(kableExtra.auto_format = FALSE)
```


# Metadata and phyloseq object bacteria {#metadata}
```{r phyloseq, warning=FALSE, message=FALSE, echo=FALSE, cache = TRUE}
#https://forum.qiime2.org/t/tutorial-integrating-qiime2-and-r-for-data-visualization-and-analysis-using-qiime2r/4121?u=jbisanz
metadata<-read_tsv("~/Documents/Study/LaTrobe/Research/phD/Jin_Project2_Sand/Jincollaboration/metadata.tsv") #requires tidyverse
metadata2 <- metadata[c(2:49),]
#defining factors
metadata2$Soils <- factor(metadata2$Soils, levels=c("Vertisol", "Chromosol"))
metadata2$CO2 <- factor(metadata2$CO2, levels=c("300", "500"))
metadata2$days <-  factor(metadata2$days, levels=c("3d", "14d", "42d"))
metadata2$Rep <-  factor(metadata2$Rep, levels=c("I", "II", "III", "IV"))

SVs      <-   qiime2R::read_qza("~/Documents/Study/LaTrobe/Research/phD/Jin_Project2_Sand/Jincollaboration/qiime2files_200317/feature_table.qza")

taxonomy <- read_tsv("~/Documents/Study/LaTrobe/Research/phD/Jin_Project2_Sand/Jincollaboration/qiime2files_200317/taxonomoy_silva.tsv")  %>% rename(Feature.ID = `Feature ID`)

taxtable <- taxonomy %>% as_tibble() %>% separate(Taxon, sep=";", c("Kingdom","Phylum","Class","Order","Family","Genus","Species")) 

tree <- read_qza("~/Documents/Study/LaTrobe/Research/phD/Jin_Project2_Sand/Jincollaboration/qiime2files_200317/insertion-tree.qza")

physeqB <- phyloseq(
  otu_table(SVs$data, taxa_are_rows = T), 
  phy_tree(tree$data), 
  tax_table(as.data.frame(taxtable) %>% dplyr::select(-Confidence) %>% column_to_rownames("Feature.ID") %>% as.matrix()), 
  sample_data(metadata2 %>% as.data.frame() %>% column_to_rownames("#SampleID")))

#adding diversity measures (with ps object) to metadata and re-create ps object 
#here a decision has to be made on sample size. Look at feature-table.qzv to see how many samples are lost with x sample size or how many features are lost with minimum sample size. This is used ONLY for diversity metrics (not differentially abundance anaylyis which I filter separately. So I generally use the minimum samples size if different between min/max is within 10fold. 
print(paste("The minimum number of sequences in this data set is",min(colSums(otu_table(physeqB))),"and the maximum number of sequences in this data set is",max(colSums(otu_table(physeqB))) ))
#Create rarefied physeq object
rare_Bac <- rarefy_even_depth(physeqB, sample.size = min(colSums(otu_table(physeqB))),  rngseed = TRUE, replace = TRUE, trimOTUs = TRUE)
ps.div <- estimate_richness(rare_Bac) #do diversity measurements and add to metadata
ps.div$`#SampleID` <- sample_names(physeqB)

meta.df <- data.frame(sample_data(physeqB))
meta.df <- meta.df %>% rownames_to_column("#SampleID") %>% 
  left_join(ps.div, by = "#SampleID") %>% column_to_rownames("#SampleID")
meta.df$pielou_ev <- ps.div$Shannon/log(ps.div$Observed)      #Pielou eveness
physeqB@sam_data <- sample_data(meta.df)

physeqB
#kable(head(sample_data(physeqB)))
#head(otu_table(physeqB))
#head(tax_table(physeqB))
kable(meta.df[c(1:5),c(1:12)], booktabs=TRUE) %>% 
  kable_styling(latex_options="scale_down")
```

# Data inspection and filtering {#inspec}
```{r inspec, warning=FALSE, message=FALSE, echo=FALSE, cache = TRUE}
# all features
otus.table <- as.data.frame(otu_table(physeqB))
otus.table <- otus.table %>% rownames_to_column(var = "#SampleID") %>% arrange(desc(rowSums(otus.table))) %>% column_to_rownames(var = "#SampleID")
#filter 
#minimum of reads per feature
physeqB.flt = prune_taxa(taxa_sums(physeqB) >= 25, physeqB) #minimum reads per feature
physeqB.flt = filter_taxa(physeqB.flt , function(x) sum(x > 0) > (0.04*length(x)), TRUE) #minimum presense in x% of samples
#filter any phyla that have not been classified i.e. are "NA"
physeqB.flt = subset_taxa(physeqB.flt, !is.na(Phylum) & !Phylum %in% c("", "uncharacterized"))

#rarefaction curve
rarecurve(t(otu_table(physeqB.flt)), step=200, sample = min(colSums(otu_table(physeqB.flt))), label = FALSE) 

#taxatables for reference
taxtableB.fltr <- data.frame(phyloseq::tax_table(physeqB.flt)@.Data)
taxtableB.fltr <- taxtableB.fltr %>% rownames_to_column(var = "Id")
# Make a data frame with a column for the read counts of each sample
sample_sum_df <- data.frame(sum = sample_sums(physeqB.flt))
summary(sample_sum_df$sum)

#histograms
# Histogram of sample read counts
ggplot(sample_sum_df, aes(x = sum)) + 
  geom_histogram(color = "black", fill = "indianred") +
  ggtitle("Distribution of sample sequencing depth") + 
  xlab("Read counts") +
  theme(axis.title.y = element_blank())

# barplot with red line through median value
dat2 <- t(otu_table(physeqB.flt)) %>% melt(.) %>% arrange(desc(value))
ggbarplot(data = dat2, x = "Var2", y = "value",xlab = "Filtered", order = unique(dat2$Var2)) + geom_hline(yintercept=median(rowSums(otu_table(physeqB.flt))), color="red")

summary(taxa_sums(physeqB.flt))
print(paste("The number of reads and ASVs left after filtering is",round(sum(taxa_sums(physeqB.flt)) / sum(taxa_sums(physeqB))*100,2),"% and",round(ntaxa(physeqB.flt) / ntaxa(physeqB) *100,2),"%, a change from",round(sum(taxa_sums(physeqB))),"to",round(sum(taxa_sums(physeqB.flt))), "reads and", ntaxa(physeqB),"to",ntaxa(physeqB.flt),"ASVs. The median changed from", median(rowSums(otu_table(physeqB))),"to",median(rowSums(otu_table(physeqB.flt))),"reads per ASV."))

otutable <- otu_table(physeqB.flt) %>% as.data.frame() %>% rownames_to_column("Feature.ID") %>% left_join(taxonomy) %>% dplyr::select( - Confidence, -Feature.ID) %>% dplyr::select(Taxon, everything())
write.csv(t(otutable), "otu.table_Jin2.csv", row.names = TRUE)
```
go [back to top](#metadata) 

# Taxa barplots
## Comparison by soil
```{r barsoil, warning=FALSE, message=FALSE, echo=FALSE, fig.cap = "taxabarplot1"}
#create plotting tables
#use relative abundances and glomerate to x level
physeqB.flt.rel.glom <- tax_glom(physeqB.flt, "Phylum")
physeqB.flt.rel.glom <-  merge_samples(physeqB.flt.rel.glom, "Soils")
physeqB.flt.rel.glom <- phyloseq::transform_sample_counts(physeqB.flt.rel.glom, function(x) x / sum(x) )
#plus relative abundances of filtered data
#use this for plots
barplot.rel <- psmelt(physeqB.flt.rel.glom) %>%  arrange(desc(Phylum))
#barplot
barplot.rel %>% ggbarplot(x = "Sample", y = "Abundance", fill = "Phylum", width = 0.2, title = "Relative Abundances of filtered OTUs", legend = "right")
```
go [back to top](#metadata) 

## Comparison by CO2
```{r barCO2, warning=FALSE, message=FALSE, echo=FALSE,cache=TRUE, fig.cap = "taxabarplot2"}
physeqBSoil1 <-  prune_samples(sample_data(physeqB.flt)$Soil == "Vertisol", physeqB.flt)
physeqBSoil2 <-  prune_samples(sample_data(physeqB.flt)$Soil == "Chromosol", physeqB.flt)

#Soil1
#use relative abundances and glomerate to x level
physeqB.flt.rel.glom <- tax_glom(physeqBSoil1, "Phylum")
physeqB.flt.rel.glom <-  merge_samples(physeqB.flt.rel.glom, "CO2")
physeqB.flt.rel.glom <- phyloseq::transform_sample_counts(physeqB.flt.rel.glom, function(x) x / sum(x) )
#barplots
barplot.rel <- psmelt(physeqB.flt.rel.glom) %>%  arrange(desc(Phylum))
barplot.rel$Sample <- factor(barplot.rel$Sample, levels=c("300", "500"))
barplot.relsoil1 <- barplot.rel %>% ggbarplot(x = "Sample", y = "Abundance", fill = "Phylum", width = 0.2, title = "Vertisol relative Abundances", legend = "right") +  theme(plot.title = element_text(size = 10))

#Soil2
physeqB.flt.rel.glom <- tax_glom(physeqBSoil2, "Phylum")
physeqB.flt.rel.glom <-  merge_samples(physeqB.flt.rel.glom, "CO2")
physeqB.flt.rel.glom <- phyloseq::transform_sample_counts(physeqB.flt.rel.glom, function(x) x / sum(x) )
#barplots
barplot.rel <- psmelt(physeqB.flt.rel.glom) %>%  arrange(desc(Phylum))
barplot.rel$Sample <- factor(barplot.rel$Sample, levels=c("300", "500"))
barplot.relsoil2 <- barplot.rel %>% ggbarplot(x = "Sample", y = "Abundance", fill = "Phylum", width = 0.2, title = "Chromosol relative Abundances", legend = "right")+  theme(plot.title = element_text(size = 10))


pdf(NULL)
gg1 <- ggarrange(barplot.relsoil2, barplot.relsoil1,  common.legend = TRUE, legend = "right")
x = dev.off()
gg1
```

```{r barCO2_class, echo=FALSE, fig.width=10, message=FALSE, warning=FALSE, cache=TRUE, fig.cap = "taxabarplot3"}
#CLASS LEVEL
#Soil1
#use relative abundances and glomerate to x level
physeqB.flt.rel.glom <- tax_glom(physeqBSoil1, "Class")
physeqB.flt.rel.glom <-  merge_samples(physeqB.flt.rel.glom, "CO2")
physeqB.flt.rel.glom <- phyloseq::transform_sample_counts(physeqB.flt.rel.glom, function(x) x / sum(x) )
#plus relative abundances of filtered data
#barplots
barplot.rel <- psmelt(physeqB.flt.rel.glom) %>%  arrange(desc(Phylum))
barplot.rel$Sample <- factor(barplot.rel$Sample, levels=c("300", "500"))
barplot.relsoil1 <- barplot.rel %>% ggbarplot(x = "Sample", y = "Abundance", fill = "Class", width = 0.2, title = "Vertisol relative Abundances", legend = "right") +  theme(plot.title = element_text(size = 10))

#Soil2
physeqB.flt.rel.glom <- tax_glom(physeqBSoil2, "Class")
physeqB.flt.rel.glom <-  merge_samples(physeqB.flt.rel.glom, "CO2")
physeqB.flt.rel.glom <- phyloseq::transform_sample_counts(physeqB.flt.rel.glom, function(x) x / sum(x) )
#plus relative abundances of filtered data
#barplots
barplot.rel <- psmelt(physeqB.flt.rel.glom) %>%  arrange(desc(Phylum))
barplot.rel$Sample <- factor(barplot.rel$Sample, levels=c("300", "500"))
barplot.relsoil2 <- barplot.rel %>% ggbarplot(x = "Sample", y = "Abundance", fill = "Class", width = 0.2, title = "Chromosol relative Abundances", legend = "right")+  theme(plot.title = element_text(size = 10))
pdf(NULL)
gg1 <- ggarrange(barplot.relsoil2, barplot.relsoil1,  common.legend = TRUE, legend = "right")
x = dev.off()
gg1
```

<br/>
go [back to top](#metadata) 

## Comparison by days and CO2 concentration
```{r bardays, echo=FALSE, fig.width=10, message=FALSE, warning=FALSE, cache=TRUE, fig.cap = "taxabarplot4"}
#Phylum level
#use relative abundances and glomerate to x level
physeqB.flt.rel.glom <- tax_glom(physeqB.flt, "Phylum")
physeqB.flt.rel.glom <- phyloseq::transform_sample_counts(physeqB.flt.rel.glom, function(x) x / sum(x) )
#plus relative abundances of filtered data
#barplots
barplot.rel <- psmelt(physeqB.flt.rel.glom) %>%  arrange(desc(Phylum))
#barplot.rel$Sample <- factor(barplot.rel$Sample, levels=c("3d", "14d", "42d"))

barplot.relsoil1 <- barplot.rel %>% ggbarplot(x = "CO2", y = "Abundance", fill = "Phylum", width = 0.2, title = "Relative Abundances of filtered ASVs", legend = "right", facet.by = c("Soils","days")) +  theme(plot.title = element_text(size = 10))
barplot.relsoil1

#Class level
#use relative abundances and glomerate to x level
physeqB.flt.rel.glom <- tax_glom(physeqB.flt, "Class")
physeqB.flt.rel.glom <- phyloseq::transform_sample_counts(physeqB.flt.rel.glom, function(x) x / sum(x) )
#barplots
barplot.rel <- psmelt(physeqB.flt.rel.glom) %>%  arrange(desc(Phylum))
barplot.relsoil1 <- barplot.rel %>% ggbarplot(x = "CO2", y = "Abundance", fill = "Class", width = 0.2, title = "Relative Abundances of filtered ASVs", legend = "right", facet.by = c("Soils","days")) +  theme(plot.title = element_text(size = 10))
barplot.relsoil1

```



# Vertisol - Sign. diff. on genus level

```{r vertsigdiff, echo=FALSE, fig.width=10, message=FALSE, warning=FALSE, eval=FALSE, include=FALSE}
#Soil1 data
physeqB.flt <- prune_taxa(taxa_sums(physeqB) >= 10, physeqB)
physeqB.flt <-  prune_samples(sample_data(physeqB.flt)$Soil == "Vertisol", physeqB.flt)
physeqtest <- tax_glom(physeqB.flt, "Genus")
physeqtest <-  prune_samples(sample_data(physeqtest)$Soil == "Vertisol", physeqtest)
physeqtest  <- filter_taxa(physeqtest, function(x) sum(x > 0) > (0.25*length(x)), TRUE) #minimum presense in x% of samples
sample_sums(physeqtest)
physeqtest <- microbiome::transform(physeqtest, "clr")
#Soil1
#wilcoxon test
physeqtest.melt <- psmelt(physeqtest) %>%  arrange(desc(Phylum))
wilcoxotest <-  physeqtest.melt %>% compare_means(Abundance ~ CO2, data = ., method = "wilcox.test", paired = FALSE, group.by = "Genus", ref.group = NULL, symnum.args = list(), p.adjust.method = "holm") %>% as.data.frame() %>% filter(p.adj <= 0.05)
kable(wilcoxotest %>% filter(Genus != "") %>% filter(Genus != "D_5__metagenome"),booktabs=TRUE) %>% kable_styling(latex_options="scale_down")
```

## Deseq2
```{r Vdeseq2, echo=FALSE, fig.width=10, message=FALSE, warning=FALSE, fig.cap = "Deseq2 significant generea", cache=TRUE}
#Soil1
#deseq2 test DAY 14
physeqB.flt <- prune_taxa(taxa_sums(physeqB) >= 10, physeqB)
tax_table(physeqB.flt)[, "Genus"] <- str_replace_all(tax_table(physeqB.flt)[, "Genus"], "D_5__", "")
physeqB.flt <-  prune_samples(sample_data(physeqB.flt)$Soil == "Vertisol", physeqB.flt)
physeqtest <- tax_glom(physeqB.flt, "Genus")
physeqtest <-  prune_samples(sample_data(physeqtest)$days == "14d", physeqtest)
physeqtest  = filter_taxa(physeqtest  , function(x) sum(x > 0) > (0.25*length(x)), TRUE) #minimum presense in x% of samples

#sample_sums(physeqtest)
#deseq2
require(DESeq2)
#http://bioconductor.org/packages/devel/bioc/vignettes/phyloseq/inst/doc/phyloseq-mixture-models.html

Soildds = phyloseq_to_deseq2(physeqtest, ~ CO2)
# calculate geometric means prior to estimate size factors
#gm_mean = function(x, na.rm=TRUE){
#  exp(sum(log(x[x > 0]), na.rm=na.rm) / length(x))
#}
#geoMeans = apply(counts(Soildds), 1, gm_mean)
#Soildds = estimateSizeFactors(Soildds, geoMeans = geoMeans)
#Soildds = DESeq(Soildds, fitType="local")
Soildds = DESeq(Soildds, test="Wald", fitType="parametric")  #default
#Note: The default multiple-inference correction is Benjamini-Hochberg, and occurs within the DESeq function.
res = results(Soildds)
res = res[order(res$padj, na.last=NA), ]
alpha = 0.01
sigtab = res[(res$padj < alpha), ]
sigtab1 = cbind(as(sigtab, "data.frame"), as(tax_table(physeqtest)[rownames(sigtab), ], "matrix"))




#deseq2 test DAY 42
physeqtest <- tax_glom(physeqB.flt, "Genus")
physeqtest <-  prune_samples(sample_data(physeqtest)$days == "42d", physeqtest)
physeqtest  = filter_taxa(physeqtest  , function(x) sum(x > 0) > (0.25*length(x)), TRUE) #minimum presense in x% of samples
#sample_sums(physeqtest)

#http://bioconductor.org/packages/devel/bioc/vignettes/phyloseq/inst/doc/phyloseq-mixture-models.html
Soildds = phyloseq_to_deseq2(physeqtest, ~ CO2)
# calculate geometric means prior to estimate size factors
#gm_mean = function(x, na.rm=TRUE){
#  exp(sum(log(x[x > 0]), na.rm=na.rm) / length(x))
#}
#geoMeans = apply(counts(Soildds), 1, gm_mean)
#Soildds = estimateSizeFactors(Soildds, geoMeans = geoMeans)
#Soildds = DESeq(Soildds, fitType="local")
Soildds = DESeq(Soildds, test="Wald", fitType="parametric")  #default
#Note: The default multiple-inference correction is Benjamini-Hochberg, and occurs within the DESeq function.
res = results(Soildds)
res = res[order(res$padj, na.last=NA), ]
alpha = 0.01
sigtab = res[(res$padj < alpha), ]
sigtab2 = cbind(as(sigtab, "data.frame"), as(tax_table(physeqtest)[rownames(sigtab), ], "matrix"))

sigtab <- rbind(sigtab1, sigtab2)
#as.vector(sigtab$Genus))
#wilcoxotest$Genus
#barplot
#barplot
barplot.rel   <- phyloseq::transform_sample_counts(physeqB.flt, function(x) x / sum(x) )
subs.mlt <- psmelt(barplot.rel)
subs <- subs.mlt %>% filter(Genus %in% sigtab$Genus)

barplot.relsoil1 <- subs   %>% ggbarplot(data = ., x = "Genus", y = "Abundance", add = "mean_se", facet.by = c("Soils","days"), fill = "CO2", width = 0.5, xlab = "", position = position_dodge(0.7), scales = "free") + theme(axis.text.x = element_text(angle = 45, hjust = 1))
barplot.relsoil1 

#kable(wilcoxotest %>% filter(Genus != "") %>% filter(Genus != "D_5__metagenome"),booktabs=TRUE) %>% kable_styling(latex_options="scale_down")
```

## Aldex
```{r aldex1, eval=TRUE, echo=FALSE, fig.width=15, fig.asp = 1.7, message=FALSE, warning=FALSE, cache=TRUE}

require(ALDEx2)
physeqB.flt <- prune_taxa(taxa_sums(physeqB) >= 10, physeqB)
tax_table(physeqB.flt)[, "Genus"] <- str_replace_all(tax_table(physeqB.flt)[, "Genus"], "D_5__", "")
#physeqB.flt <-  prune_samples(sample_data(physeqB.flt)$Soil == "Vertisol", physeqB.flt)
#physeqtest <- tax_glom(physeqB.flt, "Genus")
#physeqtest <-  prune_samples(sample_data(physeqtest)$days == "14d", physeqtest)
#physeqtest  = filter_taxa(physeqtest  , function(x) sum(x > 0) > (0.25*length(x)), TRUE) #minimum presense in x% of samples
#taxtable.test <- tax_table(physeqtest)@.Data %>% as.data.frame() %>% rownames_to_column("Id")
#summary(sample_data(physeqtest)$CO2)

#otutable <- as.data.frame(phyloseq::otu_table(physeqtest))
#otutable <- round(otutable)

#cond <- sample_data(physeqtest)$CO2
#x <- aldex.clr(otutable, cond, mc.samples=1000, denom="all", verbose=T)
#x.effect <- aldex.effect(x, verbose=TRUE)
#x.tt <- aldex.ttest(x, paired.test=FALSE, verbose=TRUE)
#x.all <- data.frame(x.effect, x.tt)
#saveRDS(x.all, "~/Documents/Study/LaTrobe/Research/phD/Jin_Project2_Sand/Jincollaboration/aldexVertisol/x.all_Vertisolday14")
#x.all <- readRDS("~/Documents/Study/LaTrobe/Research/phD/Jin_Project2_Sand/Jincollaboration/aldexVertisol/x.all_Vertisolday14")
#aldex.plot(x.all, type="MA", test="welch")
#aldex.plot(x.all, type="MW", test="welch")
#list the top features
#x.effect.list <- x.effect %>% rownames_to_column("Id")
#aldex_significant_features  <- x.tt %>% rownames_to_column("Id") %>%  left_join(x.effect.list) %>% inner_join(taxtable.test) %>% dplyr::select(Id, Phylum, Genus, wi.eBH, we.eBH, effect) %>% arrange(we.eBH) %>% filter(we.eBH <= 0.05)
#write.csv(aldex_significant_features, "~/Documents/Study/LaTrobe/Research/phD/Jin_Project2_Sand/Jincollaboration/aldexVertisol/aldex_significant_features_day14.csv")
aldex_significant_features1 <- read.csv("~/Documents/Study/LaTrobe/Research/phD/Jin_Project2_Sand/Jincollaboration/aldexVertisol/aldex_significant_features_day14.csv")


## day 42
#physeqtest <- tax_glom(physeqB.flt, "Genus")
#physeqtest <-  prune_samples(sample_data(physeqtest)$days == "42d", physeqtest)
#physeqtest  = filter_taxa(physeqtest  , function(x) sum(x > 0) > (0.25*length(x)), TRUE) #minimum presense in x% of samples
#taxtable.test <- tax_table(physeqtest)@.Data %>% as.data.frame() %>% rownames_to_column("Id")
#summary(sample_data(physeqtest)$CO2)

#otutable <- as.data.frame(phyloseq::otu_table(physeqtest))
#otutable <- round(otutable)

#cond <- sample_data(physeqtest)$CO2
#x <- aldex.clr(otutable, cond, mc.samples=1000, denom="all", verbose=T)
#x.effect <- aldex.effect(x, verbose=TRUE)
#x.tt <- aldex.ttest(x, paired.test=FALSE, verbose=TRUE)
#x.all <- data.frame(x.effect, x.tt)
#saveRDS(x.all, "~/Documents/Study/LaTrobe/Research/phD/Jin_Project2_Sand/Jincollaboration/aldexVertisol/x.all_Vertisolday42")
#x.all <- readRDS("~/Documents/Study/LaTrobe/Research/phD/Jin_Project2_Sand/Jincollaboration/aldexVertisol/x.all_Vertisolday42")
#aldex.plot(x.all, type="MA", test="welch")
#aldex.plot(x.all, type="MW", test="welch")
#list the top features
#x.effect.list <- x.effect %>% rownames_to_column("Id")
#aldex_significant_features  <- x.tt %>% rownames_to_column("Id") %>%  left_join(x.effect.list) %>% inner_join(taxtable.test) %>% dplyr::select(Id, Phylum, Genus, wi.eBH, we.eBH, effect) %>% arrange(we.eBH) %>% filter(we.eBH <= 0.05)
#write.csv(aldex_significant_features, "~/Documents/Study/LaTrobe/Research/phD/Jin_Project2_Sand/Jincollaboration/aldexVertisol/aldex_significant_features_day42.csv")
aldex_significant_features2 <- read.csv("~/Documents/Study/LaTrobe/Research/phD/Jin_Project2_Sand/Jincollaboration/aldexVertisol/aldex_significant_features_day42.csv")

aldex_significant_features <- rbind(aldex_significant_features1,aldex_significant_features2)
kable(aldex_significant_features %>% arrange(Genus) %>% dplyr::select(-X, -Id), booktabs=TRUE) %>% kable_styling(latex_options="scale_down")
```


  
# Chromosol - Sign. diff. on genus level
```{r chromosigdiff, eval = FALSE, include = FALSE, echo=FALSE, fig.width=10, message=FALSE, warning=FALSE, cache=TRUE}

#Soil2 data
physeqBSoil2 <-  prune_samples(sample_data(physeqB.flt)$Soil == "Chromosol", physeqB.flt)
#physeqBSoil2 = subset_taxa(physeqBSoil2, !is.na(Genus) & !Genus %in% c("", "uncharacterized"))

#Soil2
#Wilcoxon test
#barplot.rel   <- phyloseq::transform_sample_counts(physeqBSoil2, function(x) x / sum(x) )
barplot.rel <- microbiome::transform(physeqBSoil2, "clr")
barplot.rel <- psmelt(barplot.rel) %>%  arrange(desc(Phylum))
wilcoxotest <-  barplot.rel %>% compare_means(Abundance ~ CO2, data = ., method = "wilcox.test", paired = FALSE,
  group.by = "Genus", ref.group = NULL, symnum.args = list(), p.adjust.method = "holm") %>% as.data.frame() %>% filter(p.adj <= 0.1)
kable(wilcoxotest, booktabs=TRUE) %>% 
  kable_styling(latex_options="scale_down")


#barplot
barplot.rel   <- phyloseq::transform_sample_counts(physeqBSoil2, function(x) x / sum(x) )
subs <- subset_taxa(barplot.rel, Genus %in% c("D_5__Bacillus", "D_5__Skermanella","D_5__Paenibacillus","D_5__Agromyces","D_5__Acidothermus","D_5__Rubrobacter"))
subs <- psmelt(subs)
barplot.relsoil2 <- subs   %>% ggbarplot(data = ., x = "Genus", y = "Abundance", add = "mean_se", facet.by = c("Soils","days"), fill = "CO2", width = 0.5, xlab = "", position = position_dodge(0.7), scales = "free") + theme(axis.text.x = element_text(angle = 45, hjust = 1))
barplot.relsoil2 
```

## Deseq2
```{r Cdeseq2, echo=FALSE, fig.width=10, message=FALSE, warning=FALSE, fig.cap = "Deseq2 significant generea", cache=TRUE}
#Soil1
#deseq2 test DAY 14
physeqB.flt <- prune_taxa(taxa_sums(physeqB) >= 10, physeqB)
tax_table(physeqB.flt)[, "Genus"] <- str_replace_all(tax_table(physeqB.flt)[, "Genus"], "D_5__", "")
physeqB.flt <-  prune_samples(sample_data(physeqB.flt)$Soil == "Chromosol", physeqB.flt)
physeqtest <- tax_glom(physeqB.flt, "Genus")
physeqtest <-  prune_samples(sample_data(physeqtest)$days == "14d", physeqtest)
physeqtest  = filter_taxa(physeqtest  , function(x) sum(x > 0) > (0.25*length(x)), TRUE) #minimum presense in x% of samples

summary(sample_data(physeqtest)$CO2)
#deseq2
require(DESeq2)
#http://bioconductor.org/packages/devel/bioc/vignettes/phyloseq/inst/doc/phyloseq-mixture-models.html

Soildds = phyloseq_to_deseq2(physeqtest, ~ CO2)
# calculate geometric means prior to estimate size factors
#gm_mean = function(x, na.rm=TRUE){
#  exp(sum(log(x[x > 0]), na.rm=na.rm) / length(x))
#}
#geoMeans = apply(counts(Soildds), 1, gm_mean)
#Soildds = estimateSizeFactors(Soildds, geoMeans = geoMeans)
#Soildds = DESeq(Soildds, fitType="local")
Soildds = DESeq(Soildds, test="Wald", fitType="parametric")  #default
#Note: The default multiple-inference correction is Benjamini-Hochberg, and occurs within the DESeq function.
res = results(Soildds)
res = res[order(res$padj, na.last=NA), ]
alpha = 0.01
sigtab = res[(res$padj < alpha), ]
sigtab1 = cbind(as(sigtab, "data.frame"), as(tax_table(physeqtest)[rownames(sigtab), ], "matrix"))




#deseq2 test DAY 42
physeqtest <- tax_glom(physeqB.flt, "Genus")
physeqtest <-  prune_samples(sample_data(physeqtest)$days == "42d", physeqtest)
physeqtest  = filter_taxa(physeqtest  , function(x) sum(x > 0) > (0.25*length(x)), TRUE) #minimum presense in x% of samples
#sample_sums(physeqtest)

#http://bioconductor.org/packages/devel/bioc/vignettes/phyloseq/inst/doc/phyloseq-mixture-models.html
Soildds = phyloseq_to_deseq2(physeqtest, ~ CO2)
# calculate geometric means prior to estimate size factors
#gm_mean = function(x, na.rm=TRUE){
#  exp(sum(log(x[x > 0]), na.rm=na.rm) / length(x))
#}
#geoMeans = apply(counts(Soildds), 1, gm_mean)
#Soildds = estimateSizeFactors(Soildds, geoMeans = geoMeans)
#Soildds = DESeq(Soildds, fitType="local")
Soildds = DESeq(Soildds, test="Wald", fitType="parametric")  #default
#Note: The default multiple-inference correction is Benjamini-Hochberg, and occurs within the DESeq function.
res = results(Soildds)
res = res[order(res$padj, na.last=NA), ]
alpha = 0.01
sigtab = res[(res$padj < alpha), ]
sigtab2 = cbind(as(sigtab, "data.frame"), as(tax_table(physeqtest)[rownames(sigtab), ], "matrix"))

sigtab <- rbind(sigtab1, sigtab2)
#as.vector(sigtab$Genus))
#wilcoxotest$Genus
#barplot
#barplot
barplot.rel   <- phyloseq::transform_sample_counts(physeqB.flt, function(x) x / sum(x) )
subs.mlt <- psmelt(barplot.rel)
subs <- subs.mlt %>% filter(Genus %in% sigtab$Genus)
barplot.relsoil1 <- subs   %>% ggbarplot(data = ., x = "Genus", y = "Abundance", add = "mean_se", facet.by = c("Soils","days"), fill = "CO2", width = 0.5, xlab = "", position = position_dodge(0.7), scales = "free") + theme(axis.text.x = element_text(angle = 45, hjust = 1))
barplot.relsoil1 

#kable(wilcoxotest %>% filter(Genus != "") %>% filter(Genus != "D_5__metagenome"),booktabs=TRUE) %>% kable_styling(latex_options="scale_down")
```

## Aldex
```{r aldex2, eval=TRUE, echo=FALSE, fig.width=15, fig.asp = 1.7, message=FALSE, warning=FALSE, cache=TRUE}

require(ALDEx2)
physeqB.flt <- prune_taxa(taxa_sums(physeqB) >= 10, physeqB)
tax_table(physeqB.flt)[, "Genus"] <- str_replace_all(tax_table(physeqB.flt)[, "Genus"], "D_5__", "")
physeqB.flt <-  prune_samples(sample_data(physeqB.flt)$Soil == "Chromosol", physeqB.flt)
#physeqtest <- tax_glom(physeqB.flt, "Genus")
#physeqtest <-  prune_samples(sample_data(physeqtest)$days == "14d", physeqtest)
#physeqtest  = filter_taxa(physeqtest  , function(x) sum(x > 0) > (0.25*length(x)), TRUE) #minimum presense in x% of samples
#taxtable.test <- tax_table(physeqtest)@.Data %>% as.data.frame() %>% rownames_to_column("Id")
#summary(sample_data(physeqtest)$CO2)

#otutable <- as.data.frame(phyloseq::otu_table(physeqtest))
#otutable <- round(otutable)

#cond <- sample_data(physeqtest)$CO2
#x <- aldex.clr(otutable, cond, mc.samples=1000, denom="all", verbose=T)
#x.effect <- aldex.effect(x, verbose=TRUE)
#x.tt <- aldex.ttest(x, paired.test=FALSE, verbose=TRUE)
#x.all <- data.frame(x.effect, x.tt)
#saveRDS(x.all, "~/Documents/Study/LaTrobe/Research/phD/Jin_Project2_Sand/Jincollaboration/aldexChromosol/x.all_Chromosolday14")
#x.all <- readRDS("~/Documents/Study/LaTrobe/Research/phD/Jin_Project2_Sand/Jincollaboration/aldexVertisol/x.all_Vertisolday14")
#aldex.plot(x.all, type="MA", test="welch")
#aldex.plot(x.all, type="MW", test="welch")
#list the top features
#x.effect.list <- x.effect %>% rownames_to_column("Id")
#aldex_significant_features  <- x.tt %>% rownames_to_column("Id") %>%  left_join(x.effect.list) %>% inner_join(taxtable.test) %>% dplyr::select(Id, Phylum, Genus, wi.eBH, we.eBH, effect) %>% arrange(we.eBH) %>% filter(we.eBH <= 0.05)
#write.csv(aldex_significant_features, "~/Documents/Study/LaTrobe/Research/phD/Jin_Project2_Sand/Jincollaboration/aldexChromosol/aldex_significant_features_day14.csv")
aldex_significant_features1 <- read.csv("~/Documents/Study/LaTrobe/Research/phD/Jin_Project2_Sand/Jincollaboration/aldexChromosol/aldex_significant_features_day14.csv")


## day 42
#physeqB.flt <- prune_taxa(taxa_sums(physeqB) >= 10, physeqB)
#tax_table(physeqB.flt)[, "Genus"] <- str_replace_all(tax_table(physeqB.flt)[, "Genus"], "D_5__", "")
#physeqB.flt <-  prune_samples(sample_data(physeqB.flt)$Soil == "Chromosol", physeqB.flt)
#physeqtest <- tax_glom(physeqB.flt, "Genus")
#physeqtest <-  prune_samples(sample_data(physeqtest)$days == "42d", physeqtest)
#physeqtest  = filter_taxa(physeqtest  , function(x) sum(x > 0) > (0.25*length(x)), TRUE) #minimum presense in x% of samples
#taxtable.test <- tax_table(physeqtest)@.Data %>% as.data.frame() %>% rownames_to_column("Id")
#summary(sample_data(physeqtest)$CO2)

#otutable <- as.data.frame(phyloseq::otu_table(physeqtest))
#otutable <- round(otutable)

#cond <- sample_data(physeqtest)$CO2
#x <- aldex.clr(otutable, cond, mc.samples=1000, denom="all", verbose=T)
#x.effect <- aldex.effect(x, verbose=TRUE)
#x.tt <- aldex.ttest(x, paired.test=FALSE, verbose=TRUE)
#x.all <- data.frame(x.effect, x.tt)
#saveRDS(x.all, "~/Documents/Study/LaTrobe/Research/phD/Jin_Project2_Sand/Jincollaboration/aldexChromosol/x.all_Chromosolday42")
#x.all <- readRDS("~/Documents/Study/LaTrobe/Research/phD/Jin_Project2_Sand/Jincollaboration/aldexChromsol/x.all_Chromosolday42")
#aldex.plot(x.all, type="MA", test="welch")
#aldex.plot(x.all, type="MW", test="welch")
#list the top features
#x.effect.list <- x.effect %>% rownames_to_column("Id")
#aldex_significant_features  <- x.tt %>% rownames_to_column("Id") %>%  left_join(x.effect.list) %>% inner_join(taxtable.test) %>% dplyr::select(Id, Phylum, Genus, wi.eBH, we.eBH, effect) %>% arrange(we.eBH) %>% filter(we.eBH <= 0.05)
#write.csv(aldex_significant_features, "~/Documents/Study/LaTrobe/Research/phD/Jin_Project2_Sand/Jincollaboration/aldexChromosol/aldex_significant_features_day42.csv")
aldex_significant_features2 <- read.csv("~/Documents/Study/LaTrobe/Research/phD/Jin_Project2_Sand/Jincollaboration/aldexChromosol/aldex_significant_features_day42.csv")

aldex_significant_features <- rbind(aldex_significant_features1,aldex_significant_features2)
kable(aldex_significant_features %>% arrange(Genus) %>% dplyr::select(-X, -Id), booktabs=TRUE) %>% kable_styling(latex_options="scale_down")
```
go [back to top](#metadata)  

# Bacillus and Paenibacillus
```{r BacPae,  eval=TRUE, echo=FALSE, fig.width=8, fig.asp = 0.7, message=FALSE, warning=FALSE, cache=TRUE, fig.cap="Relative abundances of *Bacillus sp*. and *Paenibacillus sp*.. The abundances of these two genera were found to be significantly different in both soils at eleveated CO~2~ using Deseq2 and Aldex2 tests for abundances in day 14 or day 42"}

physeqB.flt <- prune_taxa(taxa_sums(physeqB) >= 10, physeqB)
tax_table(physeqB.flt)[, "Genus"] <- str_replace_all(tax_table(physeqB.flt)[, "Genus"], "D_5__", "")
physeqtest <- tax_glom(physeqB.flt, "Genus")
barplot.rel   <- phyloseq::transform_sample_counts(physeqtest , function(x) x / sum(x) )
subs <- subset_taxa(barplot.rel, Genus %in% c("Bacillus", "Paenibacillus"))
subs <- psmelt(subs)
barplot.relsoil1 <- subs   %>% ggbarplot(data = ., x = "Genus", y = "Abundance", add = "mean_se", facet.by = c("Soils","days"), fill = "CO2", width = 0.5, xlab = "", position = position_dodge(0.7), scales = "free") + theme(axis.text.x = element_text(angle = 45, hjust = 1))
barplot.relsoil1 
```



# Alpha diversity {#alphadiversity}
## by soil
```{r alphasoil, warning=FALSE, message=FALSE, echo=FALSE, cache=TRUE, fig.cap = "alpha diversity"}

as.data.frame(sample_data(physeqB.flt)) %>% melt(., measure.var = c("Observed", "InvSimpson", "Shannon", "pielou_ev"))  %>%
ggboxplot(., x = "Soils", y = "value", add = "jitter", facet.by = "variable", scales = "free", ylab = "alpha diversity") +
  stat_compare_means( method = "wilcox") 
```
  
go [back to top](#metadata) 


## by CO2
```{r alphaCO2, echo=FALSE, fig.height=10,fig.asp= 0.9, message=FALSE, warning=FALSE, fig.cap = "alpha diversity"}
p1 <- as.data.frame(sample_data(physeqB.flt)) %>% melt(., measure.var = c("Observed"))  %>%
ggboxplot(., x = "CO2", y = "value", add = "jitter", facet.by = c("Soils","variable"), scales = "free", ylab = "alpha diversity") + stat_compare_means( method = "wilcox") 
p2 <- as.data.frame(sample_data(physeqB.flt)) %>% melt(., measure.var = c("InvSimpson"))  %>%
ggboxplot(., x = "CO2", y = "value", add = "jitter", facet.by = c("Soils","variable"), scales = "free", ylab = "alpha diversity") + stat_compare_means( method = "wilcox") 
p3 <- as.data.frame(sample_data(physeqB.flt)) %>% melt(., measure.var = c("Shannon"))  %>%
ggboxplot(., x = "CO2", y = "value", add = "jitter", facet.by = c("Soils","variable"), scales = "free", ylab = "alpha diversity") + stat_compare_means( method = "wilcox") 
p4 <- as.data.frame(sample_data(physeqB.flt)) %>% melt(., measure.var = c("pielou_ev"))  %>%
ggboxplot(., x = "CO2", y = "value", add = "jitter", facet.by = c("Soils","variable"), scales = "free", ylab = "alpha diversity") + stat_compare_means( method = "wilcox") 

pdf(NULL)
gg1 <- ggarrange(p1,p2,p3,p4)
x = dev.off()
gg1

```
  
go [back to top](#metadata) 

## by days and CO2
```{r alphadaysV, warning=FALSE, message=FALSE, echo=FALSE, fig.height=10,fig.asp= 0.9, fig.cap = "alpha diversity"}

p1 <- as.data.frame(sample_data(physeqB.flt)) %>% melt(., measure.var = c("Observed"))  %>% filter(Soils == "Vertisol") %>% 
ggboxplot(., x = "days", y = "value", add = "jitter", facet.by = c("CO2","variable"), scales = "free", ylab = "alpha diversity") + stat_compare_means( method = "kruskal") 
p2 <- as.data.frame(sample_data(physeqB.flt)) %>% melt(., measure.var = c("InvSimpson"))   %>% filter(Soils == "Vertisol") %>% 
ggboxplot(., x = "days", y = "value", add = "jitter", facet.by = c("CO2","variable"), scales = "free", ylab = "alpha diversity") + stat_compare_means( method = "kruskal") 
p3 <- as.data.frame(sample_data(physeqB.flt)) %>% melt(., measure.var = c("Shannon"))   %>% filter(Soils == "Vertisol") %>% 
ggboxplot(., x = "days", y = "value", add = "jitter", facet.by = c("CO2","variable"), scales = "free", ylab = "alpha diversity") + stat_compare_means( method = "kruskal") 
p4 <- as.data.frame(sample_data(physeqB.flt)) %>% melt(., measure.var = c("pielou_ev"))   %>% filter(Soils == "Vertisol") %>% 
ggboxplot(., x = "days", y = "value", add = "jitter", facet.by = c("CO2","variable"), scales = "free", ylab = "alpha diversity") + stat_compare_means( method = "kruskal") 
pdf(NULL)
gg1 <- ggarrange(p1,p2,p3,p4)
gg1 <- annotate_figure(gg1, fig.lab  = "Vertisol",fig.lab.pos	= "top.left",fig.lab.size = 14)
x = dev.off()
gg1
```
  
go [back to top](#metadata) 

```{r alphadaysC, echo=FALSE, message=FALSE, warning=FALSE,fig.height=10,fig.asp= 0.9, fig.cap = "alpha diversity"}

p1 <- as.data.frame(sample_data(physeqB.flt)) %>% melt(., measure.var = c("Observed"))  %>% filter(Soils == "Chromosol") %>% 
ggboxplot(., x = "days", y = "value", add = "jitter", facet.by = c("CO2","variable"), scales = "free", ylab = "alpha diversity") + stat_compare_means( method = "kruskal") 
p2 <- as.data.frame(sample_data(physeqB.flt)) %>% melt(., measure.var = c("InvSimpson"))   %>% filter(Soils == "Chromosol") %>% 
ggboxplot(., x = "days", y = "value", add = "jitter", facet.by = c("CO2","variable"), scales = "free", ylab = "alpha diversity") + stat_compare_means( method = "kruskal") 
p3 <- as.data.frame(sample_data(physeqB.flt)) %>% melt(., measure.var = c("Shannon"))   %>% filter(Soils == "Chromosol") %>% 
ggboxplot(., x = "days", y = "value", add = "jitter", facet.by = c("CO2","variable"), scales = "free", ylab = "alpha diversity") + stat_compare_means( method = "kruskal") 
p4 <- as.data.frame(sample_data(physeqB.flt)) %>% melt(., measure.var = c("pielou_ev"))   %>% filter(Soils == "Chromosol") %>% 
ggboxplot(., x = "days", y = "value", add = "jitter", facet.by = c("CO2","variable"), scales = "free", ylab = "alpha diversity") + stat_compare_means( method = "kruskal") 
pdf(NULL)
gg1 <- ggarrange(p1,p2,p3,p4)
gg1 <- annotate_figure(gg1, fig.lab  = "Chromosol",fig.lab.pos	= "top.left",fig.lab.size = 14)
x = dev.off()
gg1
```
  
<br/>

(Fig. \@ref(fig:alphadaysC))  
The alpha diversity in the Chromosol was more sensitive to elevated CO~2~ compared to the Vertisol. Over time the number of ASVs reduced and this was most significant in the Chromosol and at elevated CO~2~. In the Chromosol, lower ASV numbers after 3 days occured along with greater diversity (Shannon, InvSimpson and Pilou Evenness) but exclusively at elevated CO~2~ concentrations. At ambient CO~2~ no changes in alpha diversity were observed in this soil.   
  
In the Vertisol diversity increased after 3 days under both, ambient and elevated CO~2~ concentrations but after 14 days the diversity under elevated CO~2~ maintained a higher diversity compared to ambient CO~2~. 

<br/>

go [back to top](#metadata) 


# Beta diversity 
## Permanova  {#perm}
```{r permanova, warning=FALSE, message=FALSE, echo=FALSE}
ps.perm <- rarefy_even_depth(physeqB.flt, sample.size = min(colSums(otu_table(physeqB.flt))),  rngseed = TRUE, replace = TRUE, trimOTUs = TRUE)
ps.perm <- phyloseq::transform_sample_counts(ps.perm, function(x) x / sum(x) )
otutable <- as.data.frame(phyloseq::otu_table(ps.perm))
otutable <- as.data.frame(t(otutable))
meta.df <- data.frame(sample_data(ps.perm))

require(vegan)
perm.soil <- adonis(otutable ~ Soils*CO2, data = meta.df ,method = "bray") 

perma.df <- data.frame(perm.soil$aov.tab[1:3,5:6])

kable(perma.df , booktabs=TRUE, digits = 2)
#perma.df  <- ggtexttable(format(perma.df, digits = 1))
#perma.df <- annotate_figure(perma.df , top =  "PERMANOVAs (filtered and rarefied ASVs, bray curtis)")
#perma.df
```

  
go [back to top](#metadata) 

## NMDS Bray
```{r nmds, message=FALSE, warning=FALSE, echo=FALSE, results=FALSE, cache = TRUE, fig.cap = "beta diversity"}
ps.perm <- rarefy_even_depth(physeqB.flt, sample.size = min(colSums(otu_table(physeqB.flt))),  rngseed = TRUE, replace = TRUE, trimOTUs = TRUE)
ps.perm <- phyloseq::transform_sample_counts(ps.perm, function(x) x / sum(x) )
ord_nmds <- phyloseq::ordinate(ps.perm, "NMDS", "bray")

nmds1 <- phyloseq::plot_ordination(ps.perm, ord_nmds , color = "Soils", shape = "CO2") + geom_point(aes(size = InvSimpson)) + ggtitle("nMDS, bray") + labs(subtitle ="filtered and rarefied OTU table")
nmds1  <- nmds1  +  geom_text_repel(label = nmds1$data$days)
nmds1 <- nmds1 + stat_ellipse(aes(group = days), type = "t", linetype = 2)
#nmds1 <- nmds1 + geom_polygon(aes(alpha = 0.6, fill = Rhizosphere,group = TreatmentID), stat = "identity") + guides(alpha = F)
nmds1 

```
  
go [back to top](#metadata) 
