---
title: "Jin Project 2"
subtitle: "Elevated CO2 and phytate"
output: 
  bookdown::html_document2:
    fig_caption: yes
    toc: yes
    toc_float: true
    number_sections: false
always_allow_html: true
editor_options: 
  chunk_output_type: console
knit: (function(input_file, encoding) {
  out_dir <- 'docs';
  rmarkdown::render(input_file,
  encoding=encoding,
  output_file=file.path(dirname(input_file), out_dir, 'index.html'))})
  
---

```{r packages, message=FALSE, warning=FALSE, echo=FALSE}
#mywd
#knit: (function(input_file, encoding) {
#  out_dir <- 'docs';
#  rmarkdown::render(input_file,
# encoding=encoding,
# output_file=file.path(dirname(input_file), out_dir, 'index.html'))})
#https://resources.github.com/whitepapers/github-and-rstudio/ #info on github page creation
setwd("~/Documents/Study/LaTrobe/Research/phD/Jin_Project2_Sand/Jincollaboration")
#mypackages
library(ggrepel)
library(qiime2R)
library(tidyverse)
library(stringr)
library(ggplot2)
theme_set(theme_bw())  
library(vegan)
library(ggpubr)
library(RColorBrewer)
library(phyloseq)
library(psych)
library(reshape2)
library(ggfortify)
library(factoextra)
library(FactoMineR)
library(maigesPack)
library(knitr)
library(microbiome)
library(SpiecEasi)
library(bookdown)
library(tinytex)
library(jakR)
library(kableExtra)
library(DESeq2)
options(kableExtra.auto_format = FALSE)
```


# Metadata and phyloseq object bacteria {#metadata}
```{r phyloseq, warning=FALSE, message=FALSE, echo=FALSE, cache = TRUE}
#https://forum.qiime2.org/t/tutorial-integrating-qiime2-and-r-for-data-visualization-and-analysis-using-qiime2r/4121?u=jbisanz
metadata<-read_tsv("~/Documents/Study/LaTrobe/Research/phD/Jin_Project2_Sand/Jincollaboration/metadata.tsv") #requires tidyverse
metadata2 <- metadata[c(2:49),] %>%
  rownames_to_column("spl")%>%
  mutate_all(type.convert) %>%
  mutate_if(is.factor, as.character)%>%
  column_to_rownames("spl")

#defining factors
metadata2$Soils <- factor(metadata2$Soils, levels=c("Vertisol", "Chromosol"))
metadata2$CO2 <- factor(metadata2$CO2, levels=c("300", "500"))
metadata2$days <-  factor(metadata2$days, levels=c("3d", "14d", "42d"))
metadata2$Rep <-  factor(metadata2$Rep, levels=c("I", "II", "III", "IV"))

SVs      <-   qiime2R::read_qza("~/Documents/Study/LaTrobe/Research/phD/Jin_Project2_Sand/Jincollaboration/qiime2files_200317/feature_table.qza")

taxonomy <- read_tsv("~/Documents/Study/LaTrobe/Research/phD/Jin_Project2_Sand/Jincollaboration/qiime2files_200317/taxonomoy_silva.tsv") %>% filter(Taxon != "categorical") %>% dplyr::rename(Feature.ID = `Feature ID`)

taxtable <- taxonomy %>% as_tibble() %>% separate(Taxon, sep=";", c("Kingdom","Phylum","Class","Order","Family","Genus","Species")) 

tree <- read_qza("~/Documents/Study/LaTrobe/Research/phD/Jin_Project2_Sand/Jincollaboration/qiime2files_200317/insertion-tree.qza")

physeqB <- phyloseq(
  otu_table(SVs$data, taxa_are_rows = T), 
  phy_tree(tree$data), 
  tax_table(as.data.frame(taxtable) %>% dplyr::select(-Confidence) %>% column_to_rownames("Feature.ID") %>% as.matrix()), 
  sample_data(metadata2 %>% as_tibble() %>% tibble::column_to_rownames("#SampleID")))

taxtableB <- data.frame(phyloseq::tax_table(physeqB)@.Data) %>% rownames_to_column("Id")
#adding diversity measures (with ps object) to metadata and re-create ps object 
#here a decision has to be made on sample size. Look at feature-table.qzv to see how many samples are lost with x sample size or how many features are lost with minimum sample size. This is used ONLY for diversity metrics (not differentially abundance anaylyis which I filter separately. So I generally use the minimum samples size if different between min/max is within 10fold. 
print(paste("The minimum number of sequences in this data set is",min(colSums(otu_table(physeqB))),"and the maximum number of sequences in this data set is",max(colSums(otu_table(physeqB))) ))
#Create rarefied physeq object
rare_Bac <- rarefy_even_depth(physeqB, sample.size = min(colSums(otu_table(physeqB))),  rngseed = TRUE, replace = TRUE, trimOTUs = TRUE)
ps.div <- estimate_richness(rare_Bac) #do diversity measurements and add to metadata
ps.div$`#SampleID` <- sample_names(physeqB)

meta.df <- data.frame(sample_data(physeqB))
meta.df <- meta.df %>% rownames_to_column("#SampleID") %>% 
  left_join(ps.div, by = "#SampleID") %>% column_to_rownames("#SampleID")
meta.df$pielou_ev <- ps.div$Shannon/log(ps.div$Observed)      #Pielou eveness
physeqB@sam_data <- sample_data(meta.df)

physeqB



#kable(head(sample_data(physeqB)))
#head(otu_table(physeqB))
#head(tax_table(physeqB))
kable(meta.df[c(1:5),c(1:12)], booktabs=TRUE) %>% 
  kable_styling(latex_options="scale_down")
```

# Data inspection and filtering {#inspec}
```{r inspec, warning=FALSE, message=FALSE, echo=FALSE, cache = TRUE}
# all features
otus.table <- as.data.frame(otu_table(physeqB))
otus.table <- otus.table %>% rownames_to_column(var = "#SampleID") %>% arrange(desc(rowSums(otus.table))) %>% column_to_rownames(var = "#SampleID")
#filter 
#minimum of reads per feature
physeqB.flt = prune_taxa(taxa_sums(physeqB) >= 25, physeqB) #minimum reads per feature
physeqB.flt = filter_taxa(physeqB.flt , function(x) sum(x > 0) > (0.04*length(x)), TRUE) #minimum presence in x% of samples
#filter any phyla that have not been classified i.e. are "NA"
physeqB.flt = subset_taxa(physeqB.flt, !is.na(Phylum) & !Phylum %in% c("", "uncharacterized"))

taxtableB_flt <- data.frame(phyloseq::tax_table(physeqB.flt)@.Data) %>% rownames_to_column("Id")
#rarefaction curve
rarecurve(t(otu_table(physeqB.flt)), step=200, sample = min(colSums(otu_table(physeqB.flt))), label = FALSE) 

#taxatables for reference
taxtableB.fltr <- data.frame(phyloseq::tax_table(physeqB.flt)@.Data)
taxtableB.fltr <- taxtableB.fltr %>% rownames_to_column(var = "Id")
# Make a data frame with a column for the read counts of each sample
sample_sum_df <- data.frame(sum = sample_sums(physeqB.flt))
summary(sample_sum_df$sum)

#histograms
# Histogram of sample read counts
ggplot(sample_sum_df, aes(x = sum)) + 
  geom_histogram(color = "black", fill = "indianred") +
  ggtitle("Distribution of sample sequencing depth") + 
  xlab("Read counts") +
  theme(axis.title.y = element_blank())

# barplot with red line through median value
dat2 <- t(otu_table(physeqB.flt)) %>% melt(.) %>% arrange(desc(value))
ggbarplot(data = dat2, x = "Var2", y = "value",xlab = "Filtered", order = unique(dat2$Var2)) + geom_hline(yintercept=median(rowSums(otu_table(physeqB.flt))), color="red")

summary(taxa_sums(physeqB.flt))
print(paste("The number of reads and ASVs left after filtering is",round(sum(taxa_sums(physeqB.flt)) / sum(taxa_sums(physeqB))*100,2),"% and",round(ntaxa(physeqB.flt) / ntaxa(physeqB) *100,2),"%, a change from",round(sum(taxa_sums(physeqB))),"to",round(sum(taxa_sums(physeqB.flt))), "reads and", ntaxa(physeqB),"to",ntaxa(physeqB.flt),"ASVs. The median changed from", median(rowSums(otu_table(physeqB))),"to",median(rowSums(otu_table(physeqB.flt))),"reads per ASV."))

otutable <- otu_table(physeqB.flt) %>% as.data.frame() %>% rownames_to_column("Feature.ID") %>% left_join(taxonomy) %>% dplyr::select( - Confidence, -Feature.ID) %>% dplyr::select(Taxon, everything())
write.csv(t(otutable), "otu.table_Jin2.csv", row.names = TRUE)
```
go [back to top](#metadata) 
 

<br/>
# Taxa barplots
## Comparison by soil
```{r barsoil, warning=FALSE, message=FALSE, echo=FALSE, fig.cap = "taxabarplot1"}
#create plotting tables
#use relative abundances and glomerate to x level
physeqB.flt.rel.glom <- tax_glom(physeqB.flt, "Phylum")
physeqB.flt.rel.glom <-  merge_samples(physeqB.flt.rel.glom, "Soils")
physeqB.flt.rel.glom <- phyloseq::transform_sample_counts(physeqB.flt.rel.glom, function(x) x / sum(x) )
#plus relative abundances of filtered data
#use this for plots
barplot.rel <- psmelt(physeqB.flt.rel.glom) %>%  arrange(desc(Phylum))
#barplot
barplot.rel %>% ggbarplot(x = "Sample", y = "Abundance", fill = "Phylum", width = 0.2, title = "Relative Abundances of filtered OTUs", legend = "right")
```
go [back to top](#metadata) 

## Comparison by CO2
```{r barCO2, warning=FALSE, message=FALSE, echo=FALSE, cache=TRUE, fig.cap = "taxabarplot2"}
physeqBSoil1 <-  prune_samples(sample_data(physeqB.flt)$Soil == "Vertisol", physeqB.flt)
physeqBSoil2 <-  prune_samples(sample_data(physeqB.flt)$Soil == "Chromosol", physeqB.flt)

#Soil1
#use relative abundances and glomerate to x level
physeqB.flt.rel.glom <- tax_glom(physeqBSoil1, "Phylum")
physeqB.flt.rel.glom <-  merge_samples(physeqB.flt.rel.glom, "CO2")
physeqB.flt.rel.glom <- phyloseq::transform_sample_counts(physeqB.flt.rel.glom, function(x) x / sum(x) )
#barplots
barplot.rel <- psmelt(physeqB.flt.rel.glom) %>%  arrange(desc(Phylum))
barplot.rel$Sample <- factor(barplot.rel$Sample, levels=c("300", "500"))
barplot.relsoil1 <- barplot.rel %>% ggbarplot(x = "Sample", y = "Abundance", fill = "Phylum", width = 0.2, title = "Vertisol relative Abundances", legend = "right") +  theme(plot.title = element_text(size = 10))

#Soil2
physeqB.flt.rel.glom <- tax_glom(physeqBSoil2, "Phylum")
physeqB.flt.rel.glom <-  merge_samples(physeqB.flt.rel.glom, "CO2")
physeqB.flt.rel.glom <- phyloseq::transform_sample_counts(physeqB.flt.rel.glom, function(x) x / sum(x) )
#barplots
barplot.rel <- psmelt(physeqB.flt.rel.glom) %>%  arrange(desc(Phylum))
barplot.rel$Sample <- factor(barplot.rel$Sample, levels=c("300", "500"))
barplot.relsoil2 <- barplot.rel %>% ggbarplot(x = "Sample", y = "Abundance", fill = "Phylum", width = 0.2, title = "Chromosol relative Abundances", legend = "right")+  theme(plot.title = element_text(size = 10))


pdf(NULL)
gg1 <- ggarrange(barplot.relsoil2, barplot.relsoil1,  common.legend = TRUE, legend = "right")
x = dev.off()
gg1
```

```{r barCO2class, echo=FALSE, fig.width=10, message=FALSE, warning=FALSE, cache=TRUE, fig.cap = "taxabarplot3"}

#CLASS LEVEL
#Soil1
#use relative abundances and glomerate to x level
physeqB.flt.rel.glom <- tax_glom(physeqBSoil1, "Class")
physeqB.flt.rel.glom <-  merge_samples(physeqB.flt.rel.glom, "CO2")
physeqB.flt.rel.glom <- phyloseq::transform_sample_counts(physeqB.flt.rel.glom, function(x) x / sum(x) )
#plus relative abundances of filtered data
#barplots
barplot.rel <- psmelt(physeqB.flt.rel.glom) %>%  arrange(desc(Phylum))
barplot.rel$Sample <- factor(barplot.rel$Sample, levels=c("300", "500"))
barplot.relsoil1 <- barplot.rel %>% ggbarplot(x = "Sample", y = "Abundance", fill = "Class", width = 0.2, title = "Vertisol relative Abundances", legend = "right") +  theme(plot.title = element_text(size = 10))

#Soil2
physeqB.flt.rel.glom <- tax_glom(physeqBSoil2, "Class")
physeqB.flt.rel.glom <-  merge_samples(physeqB.flt.rel.glom, "CO2")
physeqB.flt.rel.glom <- phyloseq::transform_sample_counts(physeqB.flt.rel.glom, function(x) x / sum(x) )
#plus relative abundances of filtered data
#barplots
barplot.rel <- psmelt(physeqB.flt.rel.glom) %>%  arrange(desc(Phylum))
barplot.rel$Sample <- factor(barplot.rel$Sample, levels=c("300", "500"))
barplot.relsoil2 <- barplot.rel %>% ggbarplot(x = "Sample", y = "Abundance", fill = "Class", width = 0.2, title = "Chromosol relative Abundances", legend = "right")+  theme(plot.title = element_text(size = 10))
pdf(NULL)
gg1 <- ggarrange(barplot.relsoil2, barplot.relsoil1,  common.legend = TRUE, legend = "right")
x = dev.off()
gg1
```

<br/>
go [back to top](#metadata) 

## Comparison by days and CO2 concentration
```{r bardays, echo=FALSE, fig.width=10, message=FALSE, warning=FALSE, cache=TRUE, fig.cap = "taxabarplot4"}
#Phylum level
#use relative abundances and glomerate to x level
physeqB.flt.rel.glom <- tax_glom(physeqB.flt, "Phylum")
physeqB.flt.rel.glom <- phyloseq::transform_sample_counts(physeqB.flt.rel.glom, function(x) x / sum(x) )
#plus relative abundances of filtered data
#barplots
barplot.rel <- psmelt(physeqB.flt.rel.glom) %>%  arrange(desc(Phylum))
#barplot.rel$Sample <- factor(barplot.rel$Sample, levels=c("3d", "14d", "42d"))

barplot.relsoil1 <- barplot.rel %>% ggbarplot(x = "CO2", y = "Abundance", fill = "Phylum", width = 0.2, title = "Relative Abundances of filtered ASVs", legend = "right", facet.by = c("Soils","days")) +  theme(plot.title = element_text(size = 10))
barplot.relsoil1

#Class level
#use relative abundances and glomerate to x level
physeqB.flt.rel.glom <- tax_glom(physeqB.flt, "Class")
physeqB.flt.rel.glom <- phyloseq::transform_sample_counts(physeqB.flt.rel.glom, function(x) x / sum(x) )
#barplots
barplot.rel <- psmelt(physeqB.flt.rel.glom) %>%  arrange(desc(Phylum))
barplot.relsoil1 <- barplot.rel %>% ggbarplot(x = "CO2", y = "Abundance", fill = "Class", width = 0.2, title = "Relative Abundances of filtered ASVs", legend = "right", facet.by = c("Soils","days")) +  theme(plot.title = element_text(size = 10))
barplot.relsoil1

```

go [back to top](#metadata)  

<br/>

# Vertisol - Sign. diff. on genus level

```{r vertsigdiff, echo=FALSE, fig.width=10, message=FALSE, warning=FALSE, eval=FALSE, include=FALSE}
#Soil1 data
physeqB.flt <- prune_taxa(taxa_sums(physeqB) >= 10, physeqB)
physeqB.flt <-  prune_samples(sample_data(physeqB.flt)$Soil == "Vertisol", physeqB.flt)
physeqtest <- tax_glom(physeqB.flt, "Genus")
physeqtest <-  prune_samples(sample_data(physeqtest)$Soil == "Vertisol", physeqtest)
physeqtest  <- filter_taxa(physeqtest, function(x) sum(x > 0) > (0.25*length(x)), TRUE) #minimum presense in x% of samples
sample_sums(physeqtest)
physeqtest <- microbiome::transform(physeqtest, "clr")
#Soil1
#wilcoxon test
physeqtest.melt <- psmelt(physeqtest) %>%  arrange(desc(Phylum))
wilcoxotest <-  physeqtest.melt %>% compare_means(Abundance ~ CO2, data = ., method = "wilcox.test", paired = FALSE, group.by = "Genus", ref.group = NULL, symnum.args = list(), p.adjust.method = "holm") %>% as.data.frame() %>% filter(p.adj <= 0.05)
kable(wilcoxotest %>% filter(Genus != "") %>% filter(Genus != "D_5__metagenome"),booktabs=TRUE) %>% kable_styling(latex_options="scale_down")
```

## Deseq2
```{r Vdeseq2, echo=FALSE, fig.width=10, message=FALSE, warning=FALSE, fig.cap = "Abundance of agglomerated genera that are significantly different in day 14 or day 42 (n = 8) using Deseq2", cache=TRUE}
#Soil1
#deseq2 test DAY 14
physeqB.flt <- prune_taxa(taxa_sums(physeqB) >= 10, physeqB)
tax_table(physeqB.flt)[, "Genus"] <- str_replace_all(tax_table(physeqB.flt)[, "Genus"], "D_5__", "")
tax_table(physeqB.flt)[, "Family"] <- str_replace_all(tax_table(physeqB.flt)[, "Family"], "D_4__", "")
physeqB.flt <-  prune_samples(sample_data(physeqB.flt)$Soil == "Vertisol", physeqB.flt)
physeqtest <- tax_glom(physeqB.flt, "Genus")
#physeqtest <- tax_glom(physeqB.flt, "Family")
physeqtest <-  prune_samples(sample_data(physeqtest)$days == "14d", physeqtest)
physeqtest  = filter_taxa(physeqtest  , function(x) sum(x > 0) > (0.20*length(x)), TRUE) #minimum presense in x% of samples
#sample_sums(physeqtest)
#deseq2
require(DESeq2)
#http://bioconductor.org/packages/devel/bioc/vignettes/phyloseq/inst/doc/phyloseq-mixture-models.html

Soildds = phyloseq_to_deseq2(physeqtest, ~ CO2)
# calculate geometric means prior to estimate size factors
#gm_mean = function(x, na.rm=TRUE){
#  exp(sum(log(x[x > 0]), na.rm=na.rm) / length(x))
#}
#geoMeans = apply(counts(Soildds), 1, gm_mean)
#Soildds = estimateSizeFactors(Soildds, geoMeans = geoMeans)
#Soildds = DESeq(Soildds, fitType="local")
Soildds = DESeq(Soildds, test="Wald", fitType="parametric")  #default
#Note: The default multiple-inference correction is Benjamini-Hochberg, and occurs within the DESeq function.
res = results(Soildds)
res = res[order(res$padj, na.last=NA), ]
alpha = 0.01
sigtab = res[(res$padj < alpha), ]
sigtab1 = cbind(as(sigtab, "data.frame"), as(tax_table(physeqtest)[rownames(sigtab), ], "matrix"))


#deseq2 test DAY 42
physeqtest <- tax_glom(physeqB.flt, "Genus")
#physeqtest <- tax_glom(physeqB.flt, "Family")
physeqtest <-  prune_samples(sample_data(physeqtest)$days == "42d", physeqtest)
physeqtest  = filter_taxa(physeqtest  , function(x) sum(x > 0) > (0.20*length(x)), TRUE) #minimum presense in x% of samples
#sample_sums(physeqtest)

#http://bioconductor.org/packages/devel/bioc/vignettes/phyloseq/inst/doc/phyloseq-mixture-models.html
Soildds = phyloseq_to_deseq2(physeqtest, ~ CO2)
# calculate geometric means prior to estimate size factors
#gm_mean = function(x, na.rm=TRUE){
#  exp(sum(log(x[x > 0]), na.rm=na.rm) / length(x))
#}
#geoMeans = apply(counts(Soildds), 1, gm_mean)
#Soildds = estimateSizeFactors(Soildds, geoMeans = geoMeans)
#Soildds = DESeq(Soildds, fitType="local")
Soildds = DESeq(Soildds, test="Wald", fitType="parametric")  #default
#Note: The default multiple-inference correction is Benjamini-Hochberg, and occurs within the DESeq function.
res = results(Soildds)
res = res[order(res$padj, na.last=NA), ]
alpha = 0.01
sigtab = res[(res$padj < alpha), ]
sigtab2 = cbind(as(sigtab, "data.frame"), as(tax_table(physeqtest)[rownames(sigtab), ], "matrix"))

sigtab <- rbind(sigtab1, sigtab2)
#as.vector(sigtab$Genus))
#wilcoxotest$Genus
#barplot
#barplot
barplot.rel   <- phyloseq::transform_sample_counts(physeqB.flt, function(x) x / sum(x) )
subs.mlt <- psmelt(barplot.rel)
subs <- subs.mlt %>% filter(Genus %in% sigtab$Genus)
#subs <- subs.mlt %>% filter(Family %in% sigtab$Family)

barplot.relsoil1 <- subs   %>% ggbarplot(data = ., x = "Genus", y = "Abundance", add = "mean_se", facet.by = c("Soils","days"), fill = "CO2", width = 0.5, xlab = "", position = position_dodge(0.7), scales = "free") + theme(axis.text.x = element_text(angle = 45, hjust = 1))
barplot.relsoil1 

#kable(wilcoxotest %>% filter(Genus != "") %>% filter(Genus != "D_5__metagenome"),booktabs=TRUE) %>% kable_styling(latex_options="scale_down")
```

## Aldex
```{r aldex1, eval=TRUE, echo=FALSE, fig.width=15, fig.asp = 1.7, message=FALSE, warning=FALSE, cache=TRUE}

require(ALDEx2)
physeqB.flt <- prune_taxa(taxa_sums(physeqB) >= 10, physeqB)
tax_table(physeqB.flt)[, "Genus"] <- str_replace_all(tax_table(physeqB.flt)[, "Genus"], "D_5__", "")
#physeqB.flt <-  prune_samples(sample_data(physeqB.flt)$Soil == "Vertisol", physeqB.flt)
#physeqtest <- tax_glom(physeqB.flt, "Genus")
#physeqtest <-  prune_samples(sample_data(physeqtest)$days == "14d", physeqtest)
#physeqtest  = filter_taxa(physeqtest  , function(x) sum(x > 0) > (0.25*length(x)), TRUE) #minimum presense in x% of samples
#taxtable.test <- tax_table(physeqtest)@.Data %>% as.data.frame() %>% rownames_to_column("Id")
#summary(sample_data(physeqtest)$CO2)

#otutable <- as.data.frame(phyloseq::otu_table(physeqtest))
#otutable <- round(otutable)

#cond <- sample_data(physeqtest)$CO2
#x <- aldex.clr(otutable, cond, mc.samples=1000, denom="all", verbose=T)
#x.effect <- aldex.effect(x, verbose=TRUE)
#x.tt <- aldex.ttest(x, paired.test=FALSE, verbose=TRUE)
#x.all <- data.frame(x.effect, x.tt)
#saveRDS(x.all, "~/Documents/Study/LaTrobe/Research/phD/Jin_Project2_Sand/Jincollaboration/aldexVertisol/x.all_Vertisolday14")
#x.all <- readRDS("~/Documents/Study/LaTrobe/Research/phD/Jin_Project2_Sand/Jincollaboration/aldexVertisol/x.all_Vertisolday14")
#aldex.plot(x.all, type="MA", test="welch")
#aldex.plot(x.all, type="MW", test="welch")
#list the top features
#x.effect.list <- x.effect %>% rownames_to_column("Id")
#aldex_significant_features  <- x.tt %>% rownames_to_column("Id") %>%  left_join(x.effect.list) %>% inner_join(taxtable.test) %>% dplyr::select(Id, Phylum, Genus, wi.eBH, we.eBH, effect) %>% arrange(we.eBH) %>% filter(we.eBH <= 0.05)
#write.csv(aldex_significant_features, "~/Documents/Study/LaTrobe/Research/phD/Jin_Project2_Sand/Jincollaboration/aldexVertisol/aldex_significant_features_day14.csv")
aldex_significant_features1 <- read.csv("~/Documents/Study/LaTrobe/Research/phD/Jin_Project2_Sand/Jincollaboration/aldexVertisol/aldex_significant_features_day14.csv")


## day 42
#physeqtest <- tax_glom(physeqB.flt, "Genus")
#physeqtest <-  prune_samples(sample_data(physeqtest)$days == "42d", physeqtest)
#physeqtest  = filter_taxa(physeqtest  , function(x) sum(x > 0) > (0.25*length(x)), TRUE) #minimum presense in x% of samples
#taxtable.test <- tax_table(physeqtest)@.Data %>% as.data.frame() %>% rownames_to_column("Id")
#summary(sample_data(physeqtest)$CO2)

#otutable <- as.data.frame(phyloseq::otu_table(physeqtest))
#otutable <- round(otutable)

#cond <- sample_data(physeqtest)$CO2
#x <- aldex.clr(otutable, cond, mc.samples=1000, denom="all", verbose=T)
#x.effect <- aldex.effect(x, verbose=TRUE)
#x.tt <- aldex.ttest(x, paired.test=FALSE, verbose=TRUE)
#x.all <- data.frame(x.effect, x.tt)
#saveRDS(x.all, "~/Documents/Study/LaTrobe/Research/phD/Jin_Project2_Sand/Jincollaboration/aldexVertisol/x.all_Vertisolday42")
#x.all <- readRDS("~/Documents/Study/LaTrobe/Research/phD/Jin_Project2_Sand/Jincollaboration/aldexVertisol/x.all_Vertisolday42")
#aldex.plot(x.all, type="MA", test="welch")
#aldex.plot(x.all, type="MW", test="welch")
#list the top features
#x.effect.list <- x.effect %>% rownames_to_column("Id")
#aldex_significant_features  <- x.tt %>% rownames_to_column("Id") %>%  left_join(x.effect.list) %>% inner_join(taxtable.test) %>% dplyr::select(Id, Phylum, Genus, wi.eBH, we.eBH, effect) %>% arrange(we.eBH) %>% filter(we.eBH <= 0.05)
#write.csv(aldex_significant_features, "~/Documents/Study/LaTrobe/Research/phD/Jin_Project2_Sand/Jincollaboration/aldexVertisol/aldex_significant_features_day42.csv")
aldex_significant_features2 <- read.csv("~/Documents/Study/LaTrobe/Research/phD/Jin_Project2_Sand/Jincollaboration/aldexVertisol/aldex_significant_features_day42.csv")

aldex_significant_features <- rbind(aldex_significant_features1,aldex_significant_features2)
kable(aldex_significant_features %>% arrange(Genus) %>% dplyr::select(-X, -Id), booktabs=TRUE, caption = "Abundance of agglomerated genera that were significantly different in day 14 or day 42 (n = 8) using Aldex2.") %>% kable_styling(latex_options="scale_down")
```

## Selbal
```{r selbal1, eval=TRUE, echo=FALSE, message=FALSE, warning=FALSE, cache=TRUE, fig.cap= "Selection of balances that best explain the difference bettwen ambient and elevated CO2 across all timepoints (n = 24)"}
require(selbal) 
#https://htmlpreview.github.io/?https://github.com/UVic-omics/selbal/blob/master/vignettes/vignette.html
physeqB.flt <- prune_taxa(taxa_sums(physeqB) >= 10, physeqB)
tax_table(physeqB.flt)[, "Genus"] <- str_replace_all(tax_table(physeqB.flt)[, "Genus"], "D_5__", "")
tax_table(physeqB.flt)[, "Family"] <- str_replace_all(tax_table(physeqB.flt)[, "Family"], "D_4__", "")
physeqB.flt <-  prune_samples(sample_data(physeqB.flt)$Soil == "Vertisol", physeqB.flt)
#physeqtest <- tax_glom(physeqB.flt, "Genus")
#physeqtest <- tax_glom(physeqB.flt, "Family")
#physeqtest  <- filter_taxa(physeqtest  , function(x) sum(x > 0) > (0.20*length(x)), TRUE)
# Define the taxa you don't want like this:
#physeqtest = phyloseq::subset_taxa(physeqtest, !str_detect(tolower(Genus), "uncultured"))
#taxtable <- data.frame(`Genus` = (tax_table(physeqtest)@.Data[,c("Genus")]))
#taxa_names(physeqtest) <- taxtable$Genus

#x <- as.data.frame(otu_table(physeqtest))
#x <- t(x) %>% as.data.frame()
#y <- sample_data(physeqtest)$CO2

#selb.CO2 <- selbal.cv(x,y)
#saveRDS(selb.CO2 ,'~/Documents/Study/LaTrobe/Research/phD/Jin_Project2_Sand/Jincollaboration/selbal/selb.CO2_VERTISOL')
selb.CO2 <- readRDS('~/Documents/Study/LaTrobe/Research/phD/Jin_Project2_Sand/Jincollaboration/selbal/selb.CO2_VERTISOL')
#selb.CO2$accuracy.nvar
selb.CO2$var.barplot
plot(selb.CO2$global.plot)
#plot.tab(selb.CO2$cv.tab)
#selb.CO2$cv.accuracy
#summary(selb.CO2$cv.accuracy) #MSE values
#selb.CO2$global.balance #list the features
#selb.CO2$glm #regression results
#selb.CO2 <- selbal(x,y) # whole process to get the scatter plot only with all features as numerator and denominator. 
#(selb.CO2$global.plot)
```

## Phylofactor
```{r echo=FALSE,  fig.width=13, fig.asp = 0.5, message=FALSE, warning=FALSE, fig.cap= "Relative abundances which significantly changed over time and under elevated CO2 of taxa bins identifed with phylofactor (model: days+CO2, n=24)", cache=TRUE, phylfactor1, eval=TRUE}
require(phylofactor)

taxonomygg <- read_tsv("~/Documents/Study/LaTrobe/Research/phD/Jin_Project2_Sand/Jincollaboration/qiime2files_200317/taxonomy_gg.tsv") %>% dplyr::rename(Feature.ID = `Feature ID`)
taxtablegg <- taxonomygg %>% as_tibble() %>% separate(Taxon, sep=";", c("Kingdom","Phylum","Class","Order","Family","Genus","Species")) 
tree_gg <- read_qza("~/Documents/Study/LaTrobe/Research/phD/Jin_Project2_Sand/Jincollaboration/qiime2files_200317/insertion-tree_gg.qza")

physeqB_gg <- phyloseq(
  otu_table(SVs$data, taxa_are_rows = T), 
  phy_tree(tree_gg$data), 
  tax_table(as.data.frame(taxtablegg) %>% dplyr::select(-Confidence) %>% column_to_rownames("Feature.ID") %>% as.matrix()), 
  sample_data(metadata2 %>% as.data.frame() %>% column_to_rownames("#SampleID")))

physeqB_gg <- subset_taxa(physeqB_gg, (Class!=" c__Chloroplast") | is.na(Class))
physeqB_gg <- subset_taxa(physeqB_gg, (Family!=" f__mitochondria") | is.na(Family))
physeqB_gg = subset_taxa(physeqB_gg, !is.na(Phylum) & !Phylum %in% c("", " p__"))

physeqB.flt <- prune_taxa(taxa_sums(physeqB_gg) >= 10, physeqB_gg)
#tax_table(physeqB.flt)[, "Genus"] <- str_replace_all(tax_table(physeqB.flt)[, "Genus"], " g__", "")
#tax_table(physeqB.flt)[, "Family"] <- str_replace_all(tax_table(physeqB.flt)[, "Family"], " f__", "")
physeqB.flt <-  prune_samples(sample_data(physeqB.flt)$Soil == "Vertisol", physeqB.flt)
physeqB.flt <- prune_taxa(taxa_sums(physeqB.flt) >= 10, physeqB.flt)
summary(taxa_sums(physeqB.flt))
physeqtest = filter_taxa(physeqB.flt , function(x) sum(x > 0) > (0.08*length(x)), TRUE) #minimum

OTUTable <- as.data.frame(otu_table(physeqtest))  #Our OTU table. Rows are OTUs and columns are samples
CO2 <- sample_data(physeqtest)$CO2        #Our independent variable
days <- sample_data(physeqtest)$days
tree <- phyloseq::phy_tree(physeqtest)
tree <- ape::unroot(tree)
taxadf <- as.data.frame(tax_table(physeqtest)@.Data)
taxadf <- unite(taxadf, taxonomy, sep = ";", remove = TRUE, na.rm = FALSE) %>% rownames_to_column("OTU_ID")
#taxadf <- taxadf %>% filter(Feature.ID %in% rownames(OTUTable)) %>% dplyr::rename(OTU_ID = Feature.ID) %>% dplyr::rename(taxonomy = Taxon)
#all(rownames(OTUTable) == tree$tip.label)
#all(tree$tip.label == taxadf$OTU_ID)

#testing which phylogenetic edges capture variation due to both time of sampling and CO2 cc
frmla <- Data ~ days*CO2 #Must put ILR coordinates as "Data"
my.data.frame <- data.frame('days'=days,'CO2'=CO2)
#PF.mult <- PhyloFactor(OTUTable,tree,X=my.data.frame,frmla=frmla,nfactors=25)
#saveRDS(PF.mult,'~/Documents/Study/LaTrobe/Research/phD/Jin_Project2_Sand/phylofactor/PF.mult_VERTISOLdays+CO2')
PF.mult <- readRDS('~/Documents/Study/LaTrobe/Research/phD/Jin_Project2_Sand/phylofactor/PF.mult_VERTISOLdays+CO2')
#PF.mult$factors
#summary(aov(PF.mult$models[[16]]))

#smry <- pf.summary(PF.mult, taxadf,  factor=6)
#td <- pf.tidy(smry)
#td 

#factor1 *** days:CO2***
#"k__Bacteria; p__Firmicutes; c__Clostridia""

#factor4*
#"k__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Paenibacillaceae; g__Paenibacillus; s__"

#factor5*
#""k__Bacteria; p__Proteobacteria; c__Gammaproteobacteria", "k__Bacteria; p__Proteobacteria; c__Betaproteobacteria", "k__Bacteria; p__Proteobacteria; c__Alphaproteobacteria", "k__Bacteria; p__Proteobacteria;NA;NA;NA;NA;NA"

#factor6*** days*CO2***
#"k__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Bacillaceae; g__Bacillus; s__horti"

#factor7***  days*CO2**
#"k__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Paenibacillaceae;NA"

#factor10***days*CO2***
#"k__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Paenibacillaceae; g__Paenibacillus; s__"

#factor14***days*CO2***
#"k__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Paenibacillaceae; g__Paenibacillus; s__amylolyticus"

#factor15**days*CO2***
#"k__Bacteria; p__Chloroflexi; c__Thermomicrobia; o__JG30-KF-CM45; f__; g__; s__"

#factor16*days*CO2**
#"k__Bacteria; p__Actinobacteria; c__Actinobacteria; o__Actinomycetales; f__Dietziaceae; g__Dietzia; s__"

#factor23days***CO2***
#"k__Bacteria; p__Firmicutes; c__Clostridia; o__Clostridiales; f__Symbiobacteriaceae; g__Symbiobacterium; s__"

#factor25days**CO2**
#"k__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Bacillaceae; g__Bacillus; s__flexus"



#par(mfrow=c(1,2))
#plot(CO2,smry$ilr,main='Isometric log-ratio',ylab='ILR balance')
#plot(CO2,smry$MeanRatio,main='Ratio of Geometric Means',ylab='Group1/Group2')
#plot(days,smry$ilr,main='Isometric log-ratio',ylab='ILR balance')
#plot(days,smry$MeanRatio,main='Ratio of Geometric Means',ylab='Group1/Group2')

#bin.projection <- pf.BINprojection(PF.mult,factor=15, rel.abund=T, prediction=T)  
#bin.taxa <- bin.projection$otus %>% lapply(.,FUN=function(otus,tax) OTUtoTaxa(otus,tax,common.name = F),tax=taxadf)
#plot stacked relative abundances 
#par(mfrow=c(1,1))
#bin.observed.relativeAbundances <- bin.projection$Data
#require(plotrix)
#binnm <- names(bin.projection$otus)
#colori4<-c("red","orangered1", "orange","yellow", "green1","green2", "aquamarine2", "lightblue","cadetblue3", "royalblue1", "blue", "blue","darkslateblue", "purple", "magenta2", "deeppink3")
#stackpoly(t(bin.observed.relativeAbundances),stack=T,main='Predicted Rel-Abund. (Vertisol)',xaxlab = c('300', '3d','500', '300','14d','500', '300','42d', '500'),xat = c(2.5, 4.5, 6.5, 10.5, 12.5, 14.5, 18.5, 20.5, 22.5))
#legend("left", legend=rownames(bin.observed.relativeAbundances), fill=smoothColors(colori4),cex=0.6 )




###
bin.projection <- pf.BINprojection(PF.mult,factor=25, rel.abund=T)  
#bin.taxa <- bin.projection$otus %>% lapply(.,FUN=function(otus,tax) OTUtoTaxa(otus,tax,common.name = F),tax=taxadf)
#plot stacked relative abundances 
par(mfrow=c(1,2))
bin.observed.relativeAbundances <- bin.projection$Data
require(plotrix)
colori4<-c("red","orangered3", "orangered", "orange2","gold1", "greenyellow", "green1","green2", "green3", "limegreen", "aquamarine2", "lightblue","cadetblue3", "royalblue1", "blue", "blue","blue", "blue", "blue", "darkorchid4", "darkorchid2","mediumorchid1","orchid2","magenta2","deeppink3","firebrick1")
stackpoly(t(bin.observed.relativeAbundances),stack=T,main='Observed Rel-Abund. (Vertisol)',xaxlab = c('300', '3d','500', '300','14d','500', '300','42d', '500'),xat = c(2.5, 4.5, 6.5, 10.5, 12.5, 14.5, 18.5, 20.5, 22.5))
legend("left", legend=rownames(bin.observed.relativeAbundances),fill=smoothColors(colori4), cex=0.6 )

bin.projection <- pf.BINprojection(PF.mult,factor=25, rel.abund=T, prediction = T)  
bin.observed.relativeAbundances <- bin.projection$Data
stackpoly(t(bin.observed.relativeAbundances),stack=T,main='Predicted Rel-Abund. (Vertisol)',xaxlab = c('300', '3d','500', '300','14d','500', '300','42d', '500'),xat = c(2.5, 4.5, 6.5, 10.5, 12.5, 14.5, 18.5, 20.5, 22.5))
legend("left", legend=rownames(bin.observed.relativeAbundances),fill=smoothColors(colori4), cex=0.6 )

par(mfrow=c(1,1))
```

<br/>
```{r boxplotsILRVERT, eval=TRUE, echo=FALSE, fig.width=7,  fig.asp = 1.8,message=FALSE, warning=FALSE, cache=FALSE, fig.cap= "Isometric Log Ratios (ILR) of taxa groups with abundances that significantly changed over time and under elevated CO2 of taxa bins identifed with phylofactor (model days*CO2, n = 24)."}
require(phylofactor)
PF.mult <- readRDS('~/Documents/Study/LaTrobe/Research/phD/Jin_Project2_Sand/phylofactor/PF.mult_VERTISOLdays+CO2')

smry <- pf.summary(PF.mult, taxadf,  factor=1)
p1 <- ggboxplot(x = "CO2", y= "Meanratio", data = data.frame(CO2 = CO2, Meanratio = smry$ilr),ylab='ILR balance', fill = "orangered3", add = "jitter") +
  theme(plot.margin=unit(c(1, .5, 0.5, 0.5),"cm"))
#ggboxplot(x = "CO2", y= "Meanratio", data = data.frame(CO2 = CO2, Meanratio = smry$MeanRatio), main='Ratio of Geometric Means',ylab='Group1/Group2')
p1.1 <- ggboxplot(x = "days", y= "Meanratio", data = data.frame(days = days, Meanratio = smry$ilr),ylab='ILR balance', fill = "orangered3", add = "jitter")+
  theme(plot.margin=unit(c(1, .5, 0.5, 0.5),"cm"))
#ggboxplot(x = "days", y= "Meanratio", data = data.frame(days = days, Meanratio = smry$MeanRatio), main='Ratio of Geometric Means',ylab='Group1/Group2')
p1.2 <- ggarrange(p1.1, p1) 
p00.3 <- annotate_figure(p1.2, fig.lab  = "p__Firmicutes; c__Clostridia" , fig.lab.pos	= "top.left",fig.lab.size = 14)


smry <- pf.summary(PF.mult, taxadf,  factor=6)
p1 <- ggboxplot(x = "CO2", y= "Meanratio", data = data.frame(CO2 = CO2, Meanratio = smry$ilr),ylab='ILR balance', fill = "limegreen", add = "jitter") +
  theme(plot.margin=unit(c(1, .5, 0.5, 0.5),"cm"))
#ggboxplot(x = "CO2", y= "Meanratio", data = data.frame(CO2 = CO2, Meanratio = smry$MeanRatio), main='Ratio of Geometric Means',ylab='Group1/Group2')
p1.1 <- ggboxplot(x = "days", y= "Meanratio", data = data.frame(days = days, Meanratio = smry$ilr),ylab='ILR balance', fill = "limegreen", add = "jitter")+
  theme(plot.margin=unit(c(1, .5, 0.5, 0.5),"cm"))
#ggboxplot(x = "days", y= "Meanratio", data = data.frame(days = days, Meanratio = smry$MeanRatio), main='Ratio of Geometric Means',ylab='Group1/Group2')
p1.2 <- ggarrange(p1.1, p1) 
p0.3 <- annotate_figure(p1.2, fig.lab  = "f__Bacillaceae; g__Bacillus; s__horti" , fig.lab.pos	= "top.left",fig.lab.size = 14)


smry <- pf.summary(PF.mult, taxadf,  factor=7)
p1 <- ggboxplot(x = "CO2", y= "Meanratio", data = data.frame(CO2 = CO2, Meanratio = smry$ilr),ylab='ILR balance', fill = "green2", add = "jitter") +
  theme(plot.margin=unit(c(1, .5, 0.5, 0.5),"cm"))
#ggboxplot(x = "CO2", y= "Meanratio", data = data.frame(CO2 = CO2, Meanratio = smry$MeanRatio), main='Ratio of Geometric Means',ylab='Group1/Group2')
p1.1 <- ggboxplot(x = "days", y= "Meanratio", data = data.frame(days = days, Meanratio = smry$ilr),ylab='ILR balance', fill = "green2", add = "jitter")+
  theme(plot.margin=unit(c(1, .5, 0.5, 0.5),"cm"))
#ggboxplot(x = "days", y= "Meanratio", data = data.frame(days = days, Meanratio = smry$MeanRatio), main='Ratio of Geometric Means',ylab='Group1/Group2')
p1.2 <- ggarrange(p1.1, p1) 
p12.3 <- annotate_figure(p1.2, fig.lab  = "o__Bacillales; f__Paenibacillaceae;NA" , fig.lab.pos	= "top.left",fig.lab.size = 14)


smry <- pf.summary(PF.mult, taxadf,  factor=10)
p1 <- ggboxplot(x = "CO2", y= "Meanratio", data = data.frame(CO2 = CO2, Meanratio = smry$ilr),ylab='ILR balance', fill = "aquamarine2", add = "jitter") +
  theme(plot.margin=unit(c(1, .5, 0.5, 0.5),"cm"))
#ggboxplot(x = "CO2", y= "Meanratio", data = data.frame(CO2 = CO2, Meanratio = smry$MeanRatio), main='Ratio of Geometric Means',ylab='Group1/Group2')
p1.1 <- ggboxplot(x = "days", y= "Meanratio", data = data.frame(days = days, Meanratio = smry$ilr),ylab='ILR balance', fill = "aquamarine2", add = "jitter")+
  theme(plot.margin=unit(c(1, .5, 0.5, 0.5),"cm"))
#ggboxplot(x = "days", y= "Meanratio", data = data.frame(days = days, Meanratio = smry$MeanRatio), main='Ratio of Geometric Means',ylab='Group1/Group2')
p1.2 <- ggarrange(p1.1, p1) 
p1.3 <- annotate_figure(p1.2, fig.lab  = "f__Paenibacillaceae; g__Paenibacillus; s__" , fig.lab.pos	= "top.left",fig.lab.size = 14)

smry <- pf.summary(PF.mult, taxadf,  factor=14)
p1 <- ggboxplot(x = "CO2", y= "Meanratio", data = data.frame(CO2 = CO2, Meanratio = smry$ilr),ylab='ILR balance', fill = "royalblue1", add = "jitter") +
  theme(plot.margin=unit(c(1, .5, 0.5, 0.5),"cm"))
#ggboxplot(x = "CO2", y= "Meanratio", data = data.frame(CO2 = CO2, Meanratio = smry$MeanRatio), main='Ratio of Geometric Means',ylab='Group1/Group2')
p1.1 <- ggboxplot(x = "days", y= "Meanratio", data = data.frame(days = days, Meanratio = smry$ilr),ylab='ILR balance', fill = "royalblue1", add = "jitter")+
  theme(plot.margin=unit(c(1, .5, 0.5, 0.5),"cm"))
#ggboxplot(x = "days", y= "Meanratio", data = data.frame(days = days, Meanratio = smry$MeanRatio), main='Ratio of Geometric Means',ylab='Group1/Group2')
p1.2 <- ggarrange(p1.1, p1) 
p1.5 <- annotate_figure(p1.2, fig.lab  = "f__Paenibacillaceae; g__Paenibacillus; s__amylolyticus" , fig.lab.pos	= "top.left",fig.lab.size = 14)


smry <- pf.summary(PF.mult, taxadf,  factor=15)
p1 <- ggboxplot(x = "CO2", y= "Meanratio", data = data.frame(CO2 = CO2, Meanratio = smry$ilr),ylab='ILR balance', fill = "blue", add = "jitter") +
  theme(plot.margin=unit(c(1, .5, 0.5, 0.5),"cm"))
#ggboxplot(x = "CO2", y= "Meanratio", data = data.frame(CO2 = CO2, Meanratio = smry$MeanRatio), main='Ratio of Geometric Means',ylab='Group1/Group2')
p1.1 <- ggboxplot(x = "days", y= "Meanratio", data = data.frame(days = days, Meanratio = smry$ilr),ylab='ILR balance', fill = "blue", add = "jitter")+
  theme(plot.margin=unit(c(1, .5, 0.5, 0.5),"cm"))
#ggboxplot(x = "days", y= "Meanratio", data = data.frame(days = days, Meanratio = smry$MeanRatio), main='Ratio of Geometric Means',ylab='Group1/Group2')
p1.2 <- ggarrange(p1.1, p1) 
p1.4 <- annotate_figure(p1.2, fig.lab  = "c__Thermomicrobia; o__JG30-KF-CM45" , fig.lab.pos	= "top.left",fig.lab.size = 14)



smry <- pf.summary(PF.mult, taxadf,  factor=23)
p1 <- ggboxplot(x = "CO2", y= "Meanratio", data = data.frame(CO2 = CO2, Meanratio = smry$ilr),ylab='ILR balance', fill = "magenta2", add = "jitter") +
  theme(plot.margin=unit(c(1, .5, 0.5, 0.5),"cm"))
#ggboxplot(x = "CO2", y= "Meanratio", data = data.frame(CO2 = CO2, Meanratio = smry$MeanRatio), main='Ratio of Geometric Means',ylab='Group1/Group2')
p1.1 <- ggboxplot(x = "days", y= "Meanratio", data = data.frame(days = days, Meanratio = smry$ilr),ylab='ILR balance', fill = "magenta2", add = "jitter")+
  theme(plot.margin=unit(c(1, .5, 0.5, 0.5),"cm"))
#ggboxplot(x = "days", y= "Meanratio", data = data.frame(days = days, Meanratio = smry$MeanRatio), main='Ratio of Geometric Means',ylab='Group1/Group2')
p1.2 <- ggarrange(p1.1, p1) 
p2.3 <- annotate_figure(p1.2, fig.lab  = "f__Symbiobacteriaceae; g__Symbiobacterium; s__" , fig.lab.pos	= "top.left",fig.lab.size = 14)

smry <- pf.summary(PF.mult, taxadf,  factor=25)
p1 <- ggboxplot(x = "CO2", y= "Meanratio", data = data.frame(CO2 = CO2, Meanratio = smry$ilr),ylab='ILR balance', fill = "firebrick1", add = "jitter" ) +
  theme(plot.margin=unit(c(1, .5, 0.5, 0.5),"cm"))
#ggboxplot(x = "CO2", y= "Meanratio", data = data.frame(CO2 = CO2, Meanratio = smry$MeanRatio), main='Ratio of Geometric Means',ylab='Group1/Group2')
p1.1 <- ggboxplot(x = "days", y= "Meanratio", data = data.frame(days = days, Meanratio = smry$ilr),ylab='ILR balance', fill = "firebrick1", add = "jitter")+
  theme(plot.margin=unit(c(1, .5, 0.5, 0.5),"cm"))
#ggboxplot(x = "days", y= "Meanratio", data = data.frame(days = days, Meanratio = smry$MeanRatio), main='Ratio of Geometric Means',ylab='Group1/Group2')
p1.2 <- ggarrange(p1.1, p1) 
p3.3 <- annotate_figure(p1.2, fig.lab  = "f__Bacillaceae; g__Bacillus; s__flexus" , fig.lab.pos	= "top.left",fig.lab.size = 14)


pdf(NULL)
boxplots <- ggarrange(p00.3, p0.3, p12.3, p1.3, p1.5, p1.4, p2.3, p3.3, nrow = 8) 
x = dev.off()
boxplots
```

```{r phylofactortablesVERT, eval=TRUE, echo=FALSE, message=FALSE, warning=FALSE, cache=TRUE}
#Tables to show taxa which form part of important bins 
PF.mult <- readRDS('~/Documents/Study/LaTrobe/Research/phD/Jin_Project2_Sand/phylofactor/PF.mult_VERTISOLdays+CO2')
bin.projection <- pf.BINprojection(PF.mult,factor=25, rel.abund=T)  
bin.observed.relativeAbundances <- bin.projection$Data
taxtable <- tax_table(physeqtest)@.Data %>% as.data.frame %>% rownames_to_column("OTU_ID")

#factor1 *** days:CO2***
#"k__Bacteria; p__Firmicutes; c__Clostridia""

#factor6*** days*CO2***
#"k__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Bacillaceae; g__Bacillus; s__horti"

#factor7***  days*CO2**
#"k__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Paenibacillaceae;NA"

#factor10***days*CO2***
#"k__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Paenibacillaceae; g__Paenibacillus; s__"

#factor14***days*CO2***
#"k__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Paenibacillaceae; g__Paenibacillus; s__amylolyticus"

#factor15**days*CO2***
#"k__Bacteria; p__Chloroflexi; c__Thermomicrobia; o__JG30-KF-CM45; f__; g__; s__"

#factor16*days*CO2**
#"k__Bacteria; p__Actinobacteria; c__Actinobacteria; o__Actinomycetales; f__Dietziaceae; g__Dietzia; s__"

#factor23days***CO2***
#"k__Bacteria; p__Firmicutes; c__Clostridia; o__Clostridiales; f__Symbiobacteriaceae; g__Symbiobacterium; s__"

#factor25days**CO2**
#"k__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Bacillaceae; g__Bacillus; s__flexus"

#bin1 <-  data.frame(OTU_ID = matrix(unlist(bin.projection$otus[["Bin 1"]]), nrow=length(bin.projection$otus[["Bin 1"]]), byrow=T),stringsAsFactors=FALSE)
#bin1  <- bin1 %>% left_join(taxtable) %>%  dplyr::select(-OTU_ID, - Species, - Kingdom) %>% unique
#kable(bin1, booktabs=TRUE, caption = "Taxa of bin 1") %>% kable_styling(latex_options="scale_down")


#bin2 <-  data.frame(OTU_ID = matrix(unlist(bin.projection$otus[["Bin 2"]]), nrow=length(bin.projection$otus[["Bin 2"]]), byrow=T),stringsAsFactors=FALSE)
#bin2  <- bin2 %>% left_join(taxtable) %>%  dplyr::select(-OTU_ID, - Species, - Kingdom) %>% unique
#bin2  <- bin2 %>% mutate(Bin = "Bin2") %>% dplyr::select(Bin, everything())

#bin3 <-  data.frame(OTU_ID = matrix(unlist(bin.projection$otus[["Bin 3"]]), nrow=length(bin.projection$otus[["Bin 3"]]), byrow=T),stringsAsFactors=FALSE)
#bin3  <- bin3 %>% left_join(taxtable) %>%  dplyr::select(-OTU_ID, - Species) %>% unique
#bin3  <- bin3 %>% mutate(Bin = "Bin3") %>% dplyr::select(Bin, everything())

bin6 <-  data.frame(OTU_ID = matrix(unlist(bin.projection$otus[["Bin 6"]]), nrow=length(bin.projection$otus[["Bin 6"]]), byrow=T),stringsAsFactors=FALSE)
bin6  <- bin6 %>% left_join(taxtable) %>%  dplyr::select(-OTU_ID, - Species, - Kingdom) %>% unique
bin6  <- bin6 %>% mutate(Bin = "Bin6") %>% dplyr::select(Bin, everything())

bin7 <-  data.frame(OTU_ID = matrix(unlist(bin.projection$otus[["Bin 7"]]), nrow=length(bin.projection$otus[["Bin 7"]]), byrow=T),stringsAsFactors=FALSE)
bin7  <- bin7 %>% left_join(taxtable) %>%  dplyr::select(-OTU_ID, - Species, - Kingdom) %>% unique
bin7  <- bin7 %>% mutate(Bin = "Bin7") %>% dplyr::select(Bin, everything())

bin8 <-  data.frame(OTU_ID = matrix(unlist(bin.projection$otus[["Bin 8"]]), nrow=length(bin.projection$otus[["Bin 8"]]), byrow=T),stringsAsFactors=FALSE)
bin8  <- bin8 %>% left_join(taxtable) %>%  dplyr::select(-OTU_ID, - Species, - Kingdom) %>% unique
bin8  <- bin8 %>% mutate(Bin = "Bin8") %>% dplyr::select(Bin, everything())


bin10 <-  data.frame(OTU_ID = matrix(unlist(bin.projection$otus[["Bin 10"]]), nrow=length(bin.projection$otus[["Bin 10"]]), byrow=T),stringsAsFactors=FALSE)
bin10  <- bin10 %>% left_join(taxtable) %>%  dplyr::select(-OTU_ID, - Species, - Kingdom) %>% unique
bin10  <- bin10 %>% mutate(Bin = "Bin10") %>% dplyr::select(Bin, everything())


bin11 <-  data.frame(OTU_ID = matrix(unlist(bin.projection$otus[["Bin 11"]]), nrow=length(bin.projection$otus[["Bin 11"]]), byrow=T),stringsAsFactors=FALSE)
bin11  <- bin11 %>% left_join(taxtable) %>%  dplyr::select(-OTU_ID, - Species, - Kingdom) %>% unique
bin11  <- bin11 %>% mutate(Bin = "Bin11") %>% dplyr::select(Bin, everything())


#bin14 <-  data.frame(OTU_ID = matrix(unlist(bin.projection$otus[["Bin 14"]]), nrow=length(bin.projection$otus[["Bin 14"]]), byrow=T),stringsAsFactors=FALSE)
#bin14  <- bin14 %>% left_join(taxtable) %>%  dplyr::select(-OTU_ID, - Species) %>% unique
#bin14  <- bin14 %>% mutate(Bin = "Bin14") %>% dplyr::select(Bin, everything())

bin15 <-  data.frame(OTU_ID = matrix(unlist(bin.projection$otus[["Bin 15"]]), nrow=length(bin.projection$otus[["Bin 15"]]), byrow=T),stringsAsFactors=FALSE)
bin15  <- bin15 %>% left_join(taxtable) %>%  dplyr::select(-OTU_ID, - Species, - Kingdom) %>% unique
bin15  <- bin15 %>% mutate(Bin = "Bin15") %>% dplyr::select(Bin, everything())


bin16 <-  data.frame(OTU_ID = matrix(unlist(bin.projection$otus[["Bin 16"]]), nrow=length(bin.projection$otus[["Bin 16"]]), byrow=T),stringsAsFactors=FALSE)
bin16  <- bin16 %>% left_join(taxtable) %>%  dplyr::select(-OTU_ID, - Species, - Kingdom) %>% unique
bin16  <- bin16 %>% mutate(Bin = "Bin16") %>% dplyr::select(Bin, everything())


bin17 <-  data.frame(OTU_ID = matrix(unlist(bin.projection$otus[["Bin 17"]]), nrow=length(bin.projection$otus[["Bin 17"]]), byrow=T),stringsAsFactors=FALSE)
bin17  <- bin17 %>% left_join(taxtable) %>%  dplyr::select(-OTU_ID, - Species, - Kingdom) %>% unique
bin17  <- bin17 %>% mutate(Bin = "Bin17") %>% dplyr::select(Bin, everything())


bin23 <-  data.frame(OTU_ID = matrix(unlist(bin.projection$otus[["Bin 23"]]), nrow=length(bin.projection$otus[["Bin 23"]]), byrow=T),stringsAsFactors=FALSE)
bin23  <- bin23 %>% left_join(taxtable) %>%  dplyr::select(-OTU_ID, - Species, - Kingdom) %>% unique
bin23  <- bin23 %>% mutate(Bin = "Bin23") %>% dplyr::select(Bin, everything())

bin24 <-  data.frame(OTU_ID = matrix(unlist(bin.projection$otus[["Bin 24"]]), nrow=length(bin.projection$otus[["Bin 24"]]), byrow=T),stringsAsFactors=FALSE)
bin24  <- bin24 %>% left_join(taxtable) %>%  dplyr::select(-OTU_ID, - Species, - Kingdom) %>% unique
bin24  <- bin24 %>% mutate(Bin = "Bin24") %>% dplyr::select(Bin, everything())


bin25 <-  data.frame(OTU_ID = matrix(unlist(bin.projection$otus[["Bin 25"]]), nrow=length(bin.projection$otus[["Bin 25"]]), byrow=T),stringsAsFactors=FALSE)
bin25  <- bin25 %>% left_join(taxtable) %>%  dplyr::select(-OTU_ID, - Species, - Kingdom) %>% unique
bin25  <- bin25 %>% mutate(Bin = "Bin25") %>% dplyr::select(Bin, everything())


bin26 <-  data.frame(OTU_ID = matrix(unlist(bin.projection$otus[["Bin 26"]]), nrow=length(bin.projection$otus[["Bin 26"]]), byrow=T),stringsAsFactors=FALSE)
bin26  <- bin26 %>% left_join(taxtable) %>%  dplyr::select(-OTU_ID, - Species, - Kingdom) %>% unique
bin26  <- bin26 %>% mutate(Bin = "Bin26") %>% dplyr::select(Bin, everything())


bin <- rbind(bin6, bin7, bin8,bin10, bin11,bin15, bin16, bin17, bin23, bin24, bin25, bin26)
kable(bin, booktabs=TRUE, caption = "Taxa represented in bins") %>% kable_styling(latex_options="scale_down")


```



go [back to top](#metadata)  



<br/>
  
# Chromosol - Sign. diff. on genus level
```{r chromosigdiff, eval = FALSE, include = FALSE, echo=FALSE, fig.width=10, message=FALSE, warning=FALSE, cache=TRUE}

#Soil2 data
physeqBSoil2 <-  prune_samples(sample_data(physeqB.flt)$Soil == "Chromosol", physeqB.flt)
#physeqBSoil2 = subset_taxa(physeqBSoil2, !is.na(Genus) & !Genus %in% c("", "uncharacterized"))

#Soil2
#Wilcoxon test
#barplot.rel   <- phyloseq::transform_sample_counts(physeqBSoil2, function(x) x / sum(x) )
barplot.rel <- microbiome::transform(physeqBSoil2, "clr")
barplot.rel <- psmelt(barplot.rel) %>%  arrange(desc(Phylum))
wilcoxotest <-  barplot.rel %>% compare_means(Abundance ~ CO2, data = ., method = "wilcox.test", paired = FALSE,
  group.by = "Genus", ref.group = NULL, symnum.args = list(), p.adjust.method = "holm") %>% as.data.frame() %>% filter(p.adj <= 0.1)
kable(wilcoxotest, booktabs=TRUE) %>% 
  kable_styling(latex_options="scale_down")


#barplot
barplot.rel   <- phyloseq::transform_sample_counts(physeqBSoil2, function(x) x / sum(x) )
subs <- subset_taxa(barplot.rel, Genus %in% c("D_5__Bacillus", "D_5__Skermanella","D_5__Paenibacillus","D_5__Agromyces","D_5__Acidothermus","D_5__Rubrobacter"))
subs <- psmelt(subs)
barplot.relsoil2 <- subs   %>% ggbarplot(data = ., x = "Genus", y = "Abundance", add = "mean_se", facet.by = c("Soils","days"), fill = "CO2", width = 0.5, xlab = "", position = position_dodge(0.7), scales = "free") + theme(axis.text.x = element_text(angle = 45, hjust = 1))
barplot.relsoil2 
```

## Deseq2
```{r Cdeseq2, echo=FALSE, fig.width=10, message=FALSE, warning=FALSE, fig.cap = "Abundance of agglomerated genera that are significantly different in day 14 or day 42 (n = 8) using Deseq2.", cache=TRUE}
#Soil1
#deseq2 test DAY 14
physeqB.flt <- prune_taxa(taxa_sums(physeqB) >= 10, physeqB)
tax_table(physeqB.flt)[, "Genus"] <- str_replace_all(tax_table(physeqB.flt)[, "Genus"], "D_5__", "")
physeqB.flt <-  prune_samples(sample_data(physeqB.flt)$Soil == "Chromosol", physeqB.flt)
physeqtest <- tax_glom(physeqB.flt, "Genus")
physeqtest <-  prune_samples(sample_data(physeqtest)$days == "14d", physeqtest)
physeqtest  = filter_taxa(physeqtest  , function(x) sum(x > 0) > (0.25*length(x)), TRUE) #minimum presense in x% of samples

summary(sample_data(physeqtest)$CO2)
#deseq2
require(DESeq2)
#http://bioconductor.org/packages/devel/bioc/vignettes/phyloseq/inst/doc/phyloseq-mixture-models.html

Soildds = phyloseq_to_deseq2(physeqtest, ~ CO2)
# calculate geometric means prior to estimate size factors
#gm_mean = function(x, na.rm=TRUE){
#  exp(sum(log(x[x > 0]), na.rm=na.rm) / length(x))
#}
#geoMeans = apply(counts(Soildds), 1, gm_mean)
#Soildds = estimateSizeFactors(Soildds, geoMeans = geoMeans)
#Soildds = DESeq(Soildds, fitType="local")
Soildds = DESeq(Soildds, test="Wald", fitType="parametric")  #default
#Note: The default multiple-inference correction is Benjamini-Hochberg, and occurs within the DESeq function.
res = results(Soildds)
res = res[order(res$padj, na.last=NA), ]
alpha = 0.01
sigtab = res[(res$padj < alpha), ]
sigtab1 = cbind(as(sigtab, "data.frame"), as(tax_table(physeqtest)[rownames(sigtab), ], "matrix"))




#deseq2 test DAY 42
physeqtest <- tax_glom(physeqB.flt, "Genus")
physeqtest <-  prune_samples(sample_data(physeqtest)$days == "42d", physeqtest)
physeqtest  = filter_taxa(physeqtest  , function(x) sum(x > 0) > (0.25*length(x)), TRUE) #minimum presense in x% of samples
#sample_sums(physeqtest)

#http://bioconductor.org/packages/devel/bioc/vignettes/phyloseq/inst/doc/phyloseq-mixture-models.html
Soildds = phyloseq_to_deseq2(physeqtest, ~ CO2)
# calculate geometric means prior to estimate size factors
#gm_mean = function(x, na.rm=TRUE){
#  exp(sum(log(x[x > 0]), na.rm=na.rm) / length(x))
#}
#geoMeans = apply(counts(Soildds), 1, gm_mean)
#Soildds = estimateSizeFactors(Soildds, geoMeans = geoMeans)
#Soildds = DESeq(Soildds, fitType="local")
Soildds = DESeq(Soildds, test="Wald", fitType="parametric")  #default
#Note: The default multiple-inference correction is Benjamini-Hochberg, and occurs within the DESeq function.
res = results(Soildds)
res = res[order(res$padj, na.last=NA), ]
alpha = 0.01
sigtab = res[(res$padj < alpha), ]
sigtab2 = cbind(as(sigtab, "data.frame"), as(tax_table(physeqtest)[rownames(sigtab), ], "matrix"))

sigtab <- rbind(sigtab1, sigtab2)
#as.vector(sigtab$Genus))
#wilcoxotest$Genus
#barplot
#barplot
barplot.rel   <- phyloseq::transform_sample_counts(physeqB.flt, function(x) x / sum(x) )
subs.mlt <- psmelt(barplot.rel)
subs <- subs.mlt %>% filter(Genus %in% sigtab$Genus)
barplot.relsoil1 <- subs   %>% ggbarplot(data = ., x = "Genus", y = "Abundance", add = "mean_se", facet.by = c("Soils","days"), fill = "CO2", width = 0.5, xlab = "", position = position_dodge(0.7), scales = "free") + theme(axis.text.x = element_text(angle = 45, hjust = 1))
barplot.relsoil1 

#kable(wilcoxotest %>% filter(Genus != "") %>% filter(Genus != "D_5__metagenome"),booktabs=TRUE) %>% kable_styling(latex_options="scale_down")
```

## Aldex
```{r aldex2, eval=TRUE, echo=FALSE, fig.width=15, fig.asp = 1.7, message=FALSE, warning=FALSE, cache=TRUE}

require(ALDEx2)
physeqB.flt <- prune_taxa(taxa_sums(physeqB) >= 10, physeqB)
tax_table(physeqB.flt)[, "Genus"] <- str_replace_all(tax_table(physeqB.flt)[, "Genus"], "D_5__", "")
physeqB.flt <-  prune_samples(sample_data(physeqB.flt)$Soil == "Chromosol", physeqB.flt)
#physeqtest <- tax_glom(physeqB.flt, "Genus")
#physeqtest <-  prune_samples(sample_data(physeqtest)$days == "14d", physeqtest)
#physeqtest  = filter_taxa(physeqtest  , function(x) sum(x > 0) > (0.25*length(x)), TRUE) #minimum presense in x% of samples
#taxtable.test <- tax_table(physeqtest)@.Data %>% as.data.frame() %>% rownames_to_column("Id")
#summary(sample_data(physeqtest)$CO2)

#otutable <- as.data.frame(phyloseq::otu_table(physeqtest))
#otutable <- round(otutable)

#cond <- sample_data(physeqtest)$CO2
#x <- aldex.clr(otutable, cond, mc.samples=1000, denom="all", verbose=T)
#x.effect <- aldex.effect(x, verbose=TRUE)
#x.tt <- aldex.ttest(x, paired.test=FALSE, verbose=TRUE)
#x.all <- data.frame(x.effect, x.tt)
#saveRDS(x.all, "~/Documents/Study/LaTrobe/Research/phD/Jin_Project2_Sand/Jincollaboration/aldexChromosol/x.all_Chromosolday14")
#x.all <- readRDS("~/Documents/Study/LaTrobe/Research/phD/Jin_Project2_Sand/Jincollaboration/aldexVertisol/x.all_Vertisolday14")
#aldex.plot(x.all, type="MA", test="welch")
#aldex.plot(x.all, type="MW", test="welch")
#list the top features
#x.effect.list <- x.effect %>% rownames_to_column("Id")
#aldex_significant_features  <- x.tt %>% rownames_to_column("Id") %>%  left_join(x.effect.list) %>% inner_join(taxtable.test) %>% dplyr::select(Id, Phylum, Genus, wi.eBH, we.eBH, effect) %>% arrange(we.eBH) %>% filter(we.eBH <= 0.05)
#write.csv(aldex_significant_features, "~/Documents/Study/LaTrobe/Research/phD/Jin_Project2_Sand/Jincollaboration/aldexChromosol/aldex_significant_features_day14.csv")
aldex_significant_features1 <- read.csv("~/Documents/Study/LaTrobe/Research/phD/Jin_Project2_Sand/Jincollaboration/aldexChromosol/aldex_significant_features_day14.csv")


## day 42
#physeqB.flt <- prune_taxa(taxa_sums(physeqB) >= 10, physeqB)
#tax_table(physeqB.flt)[, "Genus"] <- str_replace_all(tax_table(physeqB.flt)[, "Genus"], "D_5__", "")
#physeqB.flt <-  prune_samples(sample_data(physeqB.flt)$Soil == "Chromosol", physeqB.flt)
#physeqtest <- tax_glom(physeqB.flt, "Genus")
#physeqtest <-  prune_samples(sample_data(physeqtest)$days == "42d", physeqtest)
#physeqtest  = filter_taxa(physeqtest  , function(x) sum(x > 0) > (0.25*length(x)), TRUE) #minimum presense in x% of samples
#taxtable.test <- tax_table(physeqtest)@.Data %>% as.data.frame() %>% rownames_to_column("Id")
#summary(sample_data(physeqtest)$CO2)

#otutable <- as.data.frame(phyloseq::otu_table(physeqtest))
#otutable <- round(otutable)

#cond <- sample_data(physeqtest)$CO2
#x <- aldex.clr(otutable, cond, mc.samples=1000, denom="all", verbose=T)
#x.effect <- aldex.effect(x, verbose=TRUE)
#x.tt <- aldex.ttest(x, paired.test=FALSE, verbose=TRUE)
#x.all <- data.frame(x.effect, x.tt)
#saveRDS(x.all, "~/Documents/Study/LaTrobe/Research/phD/Jin_Project2_Sand/Jincollaboration/aldexChromosol/x.all_Chromosolday42")
#x.all <- readRDS("~/Documents/Study/LaTrobe/Research/phD/Jin_Project2_Sand/Jincollaboration/aldexChromsol/x.all_Chromosolday42")
#aldex.plot(x.all, type="MA", test="welch")
#aldex.plot(x.all, type="MW", test="welch")
#list the top features
#x.effect.list <- x.effect %>% rownames_to_column("Id")
#aldex_significant_features  <- x.tt %>% rownames_to_column("Id") %>%  left_join(x.effect.list) %>% inner_join(taxtable.test) %>% dplyr::select(Id, Phylum, Genus, wi.eBH, we.eBH, effect) %>% arrange(we.eBH) %>% filter(we.eBH <= 0.05)
#write.csv(aldex_significant_features, "~/Documents/Study/LaTrobe/Research/phD/Jin_Project2_Sand/Jincollaboration/aldexChromosol/aldex_significant_features_day42.csv")
aldex_significant_features2 <- read.csv("~/Documents/Study/LaTrobe/Research/phD/Jin_Project2_Sand/Jincollaboration/aldexChromosol/aldex_significant_features_day42.csv")

aldex_significant_features <- rbind(aldex_significant_features1,aldex_significant_features2)
kable(aldex_significant_features %>% arrange(Genus) %>% dplyr::select(-X, -Id), booktabs=TRUE,caption = "Abundance of agglomerated genera that were significantly different in day 14 or day 42 (n = 8) using Aldex2.") %>% kable_styling(latex_options="scale_down")
```
go [back to top](#metadata)  

## Selbal
```{r selbal2, eval=TRUE, echo=FALSE, message=FALSE, warning=FALSE, cache=TRUE, fig.cap= "Selection of balances that best explain the difference bettwen ambient and elevated CO2 across all timepoints (n = 24)"}
require(selbal) 
#https://htmlpreview.github.io/?https://github.com/UVic-omics/selbal/blob/master/vignettes/vignette.html
physeqB.flt <- prune_taxa(taxa_sums(physeqB) >= 10, physeqB)
tax_table(physeqB.flt)[, "Genus"] <- str_replace_all(tax_table(physeqB.flt)[, "Genus"], "D_5__", "")
tax_table(physeqB.flt)[, "Family"] <- str_replace_all(tax_table(physeqB.flt)[, "Family"], "D_4__", "")
physeqB.flt <-  prune_samples(sample_data(physeqB.flt)$Soil == "Chromosol", physeqB.flt)
#physeqtest <- tax_glom(physeqB.flt, "Genus")
#physeqtest <- tax_glom(physeqB.flt, "Family")
#physeqtest  <- filter_taxa(physeqtest  , function(x) sum(x > 0) > (0.20*length(x)), TRUE)
# Define the taxa you don't want like this:
#physeqtest = phyloseq::subset_taxa(physeqtest, !str_detect(tolower(Genus), "uncultured"))
#physeqtest = phyloseq::subset_taxa(physeqtest, !str_detect(tolower(Genus), "metagenome"))
#taxtable <- data.frame(`Genus` = (tax_table(physeqtest)@.Data[,c("Genus")]))
#taxa_names(physeqtest) <- taxtable$Genus

#x <- as.data.frame(otu_table(physeqtest))
#x <- t(x) %>% as.data.frame()
#y <- sample_data(physeqtest)$CO2

#selb.CO2 <- selbal.cv(x,y)
#saveRDS(selb.CO2 ,'~/Documents/Study/LaTrobe/Research/phD/Jin_Project2_Sand/Jincollaboration/selbal/selb.CO2_CHROMOSOL')
selb.CO2 <- readRDS('~/Documents/Study/LaTrobe/Research/phD/Jin_Project2_Sand/Jincollaboration/selbal/selb.CO2_CHROMOSOL')
#se.pw.flt.gm.CN$accuracy.nvar
selb.CO2$var.barplot
plot(selb.CO2$global.plot)
#plot.tab(selb.CO2$cv.tab)
#selb.CO2$cv.accuracy
#summary(selb.CO2$cv.accuracy) #MSE values
#selb.CO2$global.balance #list the features
#selb.CO2$glm #regression results
#selb.CO2 <- selbal(x,y) # whole process to get the scatter plot only with all features as numerator and denominator. 
#(selb.CO2$global.plot)
```


## Phylofactor
```{r , phylfactor2, echo=FALSE,  fig.width=13, fig.asp = 0.5, message=FALSE, warning=FALSE, fig.cap= "Relative abundances which significantly changed over time and under elevated CO2 of taxa bins identifed with phylofactor (model days*CO2, n=24)", cache=FALSE, eval=TRUE}
require(phylofactor)

taxonomygg <- read_tsv("~/Documents/Study/LaTrobe/Research/phD/Jin_Project2_Sand/Jincollaboration/qiime2files_200317/taxonomy_gg.tsv") %>% filter(Taxon != "categorical") %>% dplyr::rename(Feature.ID = `Feature ID`)
taxtablegg <- taxonomygg %>% as_tibble() %>% separate(Taxon, sep=";", c("Kingdom","Phylum","Class","Order","Family","Genus","Species")) 
tree_gg <- read_qza("~/Documents/Study/LaTrobe/Research/phD/Jin_Project2_Sand/Jincollaboration/qiime2files_200317/insertion-tree_gg.qza")

  
physeqB_gg <- phyloseq(
  otu_table(SVs$data, taxa_are_rows = T), 
  phy_tree(tree_gg$data), 
  tax_table(as.data.frame(taxtablegg) %>% dplyr::select(-Confidence) %>% column_to_rownames("Feature.ID") %>% as.matrix()), 
  sample_data(metadata2 %>% as_tibble() %>% column_to_rownames("#SampleID")))

#filtering out mitochondria and Chloroplast and unknown phyla
#subset_taxa(physeqB_gg, Class==" c__Chloroplast")
#subset_taxa(physeqB_gg, Family == " f__mitochondria")
physeqB_gg <- subset_taxa(physeqB_gg, (Class!=" c__Chloroplast") | is.na(Class))
physeqB_gg <- subset_taxa(physeqB_gg, (Family!=" f__mitochondria") | is.na(Family))
physeqB_gg = subset_taxa(physeqB_gg, !is.na(Phylum) & !Phylum %in% c("", " p__"))

physeqB.flt <- prune_taxa(taxa_sums(physeqB_gg) >= 10, physeqB_gg)
#tax_table(physeqB.flt)[, "Genus"] <- str_replace_all(tax_table(physeqB.flt)[, "Genus"], " g__", "")
#tax_table(physeqB.flt)[, "Family"] <- str_replace_all(tax_table(physeqB.flt)[, "Family"], " f__", "")
physeqB.flt <-  prune_samples(sample_data(physeqB.flt)$Soil == "Chromosol", physeqB.flt)
physeqB.flt <- prune_taxa(taxa_sums(physeqB.flt) >= 10, physeqB.flt)
summary(taxa_sums(physeqB.flt))
physeqtest = filter_taxa(physeqB.flt , function(x) sum(x > 0) > (0.08*length(x)), TRUE) #minimum

OTUTable <- as.data.frame(otu_table(physeqtest))  #Our OTU table. Rows are OTUs and columns are samples
CO2 <- sample_data(physeqtest)$CO2        #Our independent variable
days <- sample_data(physeqtest)$days
tree <- phyloseq::phy_tree(physeqtest)
tree <- ape::unroot(tree)
taxadf <- as.data.frame(tax_table(physeqtest)@.Data)
taxadf <- unite(taxadf, taxonomy, sep = ";", remove = TRUE, na.rm = FALSE) %>% rownames_to_column("OTU_ID")
#taxadf <- taxadf %>% filter(Feature.ID %in% rownames(OTUTable)) %>% dplyr::rename(OTU_ID = Feature.ID) %>% dplyr::rename(taxonomy = Taxon)
#all(rownames(OTUTable) == tree$tip.label)
#all(tree$tip.label == taxadf$OTU_ID)

#testing which phylogenetic edges capture variation due to both time of sampling and CO2 cc
frmla <- Data ~ days*CO2 #Must put ILR coordinates as "Data"
my.data.frame <- data.frame('days'=days,'CO2'=CO2)
#PF.mult <- PhyloFactor(OTUTable,tree,X=my.data.frame,frmla=frmla,nfactors=25)
#saveRDS(PF.mult,'~/Documents/Study/LaTrobe/Research/phD/Jin_Project2_Sand/phylofactor/PF.mult_CHROMOSOLdays+CO2')
PF.mult <- readRDS('~/Documents/Study/LaTrobe/Research/phD/Jin_Project2_Sand/phylofactor/PF.mult_CHROMOSOLdays+CO2')
#PF.mult$factors
#summary(aov(PF.mult$models[[25]]))

# sig. factors for CO2 1., 3.(interaction.), 4*(interaction*), 6(interation only**), 7*(interaction***), 8*, 9**(interaction***), 10(interaction only**), 13*(interaction.), 16*, 17***(interaction***),22*(interaction.), 24***(interaction*), 25**(interaction**)

#smry <- pf.summary(PF.mult, taxadf,  factor=11)
#td <- pf.tidy(smry)
#td

#factor4
# ("k__Bacteria; p__Proteobacteria; c__Deltaproteobacteria; o__Desulfuromonadales", "k__Bacteria; p__Acidobacteria", "k__Bacteria; p__Actinobacteria", "k__Bacteria; p__Chloroflexi",  "k__Bacteria; p__Planctomycetes" , "k__Bacteria; p__Verrucomicrobia""k__Bacteria; p__Firmicutes")

#factor6
#c("k__Bacteria; p__Actinobacteria; c__Actinobacteria; o__Actinomycetales; f__Micrococcaceae; g__Arthrobacter;NA")

#factor7
#("k__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Bacillaceae; g__Bacillus; s__horti")

#factor9
# ( "k__Bacteria; p__Firmicutes; c__Clostridia; o__Clostridiales; f__Symbiobacteriaceae")

#factor17
# ("k__Bacteria; p__Actinobacteria; c__Actinobacteria; o__Actinomycetales; f__Propionibacteriaceae" )

#factor24
# ("k__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Paenibacillaceae; g__Cohnella")

#factor25
# ("k__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Paenibacillaceae;NA;NA")
#my_title <- expression(paste("Family ", italic("Propionibacteriaceae")))

bin.projection <- pf.BINprojection(PF.mult,factor=25, rel.abund=T)  
#bin.taxa <- bin.projection$otus %>% lapply(.,FUN=function(otus,tax) OTUtoTaxa(otus,tax,common.name = F),tax=taxadf)
#plot stacked relative abundances 
par(mfrow=c(1,2))
bin.observed.relativeAbundances <- bin.projection$Data
require(plotrix)
colori4<-c("red","orangered3", "orangered", "orange2","gold1", "greenyellow", "green1","green2", "green3", "limegreen", "aquamarine2", "lightblue","cadetblue3", "royalblue1", "blue", "blue","blue", "blue", "blue", "darkorchid4", "darkorchid2","mediumorchid1","orchid2","magenta2","deeppink3","firebrick1")
stackpoly(t(bin.observed.relativeAbundances),stack=T,main='Observed Rel-Abund. (Chromosol)',xaxlab = c('300', '3d','500', '300','14d','500', '300','42d', '500'),xat = c(2.5, 4.5, 6.5, 10.5, 12.5, 14.5, 18.5, 20.5, 22.5))
legend("left", legend=rownames(bin.observed.relativeAbundances),fill=smoothColors(colori4), cex=0.6 )

bin.projection <- pf.BINprojection(PF.mult,factor=25, rel.abund=T, prediction = T)  
bin.observed.relativeAbundances <- bin.projection$Data
stackpoly(t(bin.observed.relativeAbundances),stack=T,main='Predicted Rel-Abund. (Chromosol)',xaxlab = c('300', '3d','500', '300','14d','500', '300','42d', '500'),xat = c(2.5, 4.5, 6.5, 10.5, 12.5, 14.5, 18.5, 20.5, 22.5))
legend("left", legend=rownames(bin.observed.relativeAbundances),fill=smoothColors(colori4), cex=0.6 )

par(mfrow=c(1,1))




```
<br/>
```{r boxplotsILR, eval=TRUE, echo=FALSE, fig.width=7,  fig.asp = 1.5,message=FALSE, warning=FALSE, cache=FALSE, fig.cap= "Isometric Log Ratios (ILR) of taxa groups with abundances that significantly changed over time and under elevated CO2 of taxa bins identifed with phylofactor (model days*CO2, n = 24)."}
PF.mult <- readRDS('~/Documents/Study/LaTrobe/Research/phD/Jin_Project2_Sand/phylofactor/PF.mult_CHROMOSOLdays+CO2')
#PF.mult$factors
#summary(aov(PF.mult$models[[25]]))

# sig. factors for CO2 1., 3.(interaction.), 4*(interaction*), 6(interation only**), 7*(interaction***), 8*, 9**(interaction***), 10(interaction only**), 13*(interaction.), 16*, 17***(interaction***),22*(interaction.), 24***(interaction*), 25**(interaction**)

#smry <- pf.summary(PF.mult, taxadf,  factor=23)
#td <- pf.tidy(smry)
#td

#factor4
# ("k__Bacteria; p__Proteobacteria; c__Deltaproteobacteria; o__Desulfuromonadales", "k__Bacteria; p__Acidobacteria", "k__Bacteria; p__Actinobacteria", "k__Bacteria; p__Chloroflexi",  "k__Bacteria; p__Planctomycetes" , "k__Bacteria; p__Verrucomicrobia""k__Bacteria; p__Firmicutes")

#factor6
#c("k__Bacteria; p__Actinobacteria; c__Actinobacteria; o__Actinomycetales; f__Micrococcaceae; g__Arthrobacter;NA")

#factor7
#("k__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Bacillaceae; g__Bacillus; s__horti")

#factor9
# ( "k__Bacteria; p__Firmicutes; c__Clostridia; o__Clostridiales; f__Symbiobacteriaceae")

#factor17
# ("k__Bacteria; p__Actinobacteria; c__Actinobacteria; o__Actinomycetales; f__Propionibacteriaceae" )


#factor24
# ("k__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Paenibacillaceae; g__Cohnella")

#factor25
# ("k__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Paenibacillaceae;NA;NA")
#my_title <- expression(paste("Family ", italic("Propionibacteriaceae")))

smry <- pf.summary(PF.mult, taxadf,  factor=7)
p1 <- ggboxplot(x = "CO2", y= "Meanratio", data = data.frame(CO2 = CO2, Meanratio = smry$ilr),ylab='ILR balance', fill = "green2", add = "jitter") +
  theme(plot.margin=unit(c(1, .5, 0.5, 0.5),"cm"))
#ggboxplot(x = "CO2", y= "Meanratio", data = data.frame(CO2 = CO2, Meanratio = smry$MeanRatio), main='Ratio of Geometric Means',ylab='Group1/Group2')
p1.1 <- ggboxplot(x = "days", y= "Meanratio", data = data.frame(days = days, Meanratio = smry$ilr),ylab='ILR balance', fill = "green2", add = "jitter")+
  theme(plot.margin=unit(c(1, .5, 0.5, 0.5),"cm"))
#ggboxplot(x = "days", y= "Meanratio", data = data.frame(days = days, Meanratio = smry$MeanRatio), main='Ratio of Geometric Means',ylab='Group1/Group2')
p1.2 <- ggarrange(p1.1, p1) 
p00.3 <- annotate_figure(p1.2, fig.lab  = "f__Bacillaceae; g__Bacillus; s__horti" , fig.lab.pos	= "top.left",fig.lab.size = 14)


smry <- pf.summary(PF.mult, taxadf,  factor=9)
p1 <- ggboxplot(x = "CO2", y= "Meanratio", data = data.frame(CO2 = CO2, Meanratio = smry$ilr),ylab='ILR balance', fill = "limegreen", add = "jitter") +
  theme(plot.margin=unit(c(1, .5, 0.5, 0.5),"cm"))
#ggboxplot(x = "CO2", y= "Meanratio", data = data.frame(CO2 = CO2, Meanratio = smry$MeanRatio), main='Ratio of Geometric Means',ylab='Group1/Group2')
p1.1 <- ggboxplot(x = "days", y= "Meanratio", data = data.frame(days = days, Meanratio = smry$ilr),ylab='ILR balance', fill = "limegreen", add = "jitter")+
  theme(plot.margin=unit(c(1, .5, 0.5, 0.5),"cm"))
#ggboxplot(x = "days", y= "Meanratio", data = data.frame(days = days, Meanratio = smry$MeanRatio), main='Ratio of Geometric Means',ylab='Group1/Group2')
p1.2 <- ggarrange(p1.1, p1) 
p0.3 <- annotate_figure(p1.2, fig.lab  = "o__Clostridiales; f__Symbiobacteriaceae" , fig.lab.pos	= "top.left",fig.lab.size = 14)


smry <- pf.summary(PF.mult, taxadf,  factor=12)
p1 <- ggboxplot(x = "CO2", y= "Meanratio", data = data.frame(CO2 = CO2, Meanratio = smry$ilr),ylab='ILR balance', fill = "green2", add = "jitter") +
  theme(plot.margin=unit(c(1, .5, 0.5, 0.5),"cm"))
#ggboxplot(x = "CO2", y= "Meanratio", data = data.frame(CO2 = CO2, Meanratio = smry$MeanRatio), main='Ratio of Geometric Means',ylab='Group1/Group2')
p1.1 <- ggboxplot(x = "days", y= "Meanratio", data = data.frame(days = days, Meanratio = smry$ilr),ylab='ILR balance', fill = "green2", add = "jitter")+
  theme(plot.margin=unit(c(1, .5, 0.5, 0.5),"cm"))
#ggboxplot(x = "days", y= "Meanratio", data = data.frame(days = days, Meanratio = smry$MeanRatio), main='Ratio of Geometric Means',ylab='Group1/Group2')
p1.2 <- ggarrange(p1.1, p1) 
p12.3 <- annotate_figure(p1.2, fig.lab  = "f__Bacillaceae; g__Bacillus; s__horti" , fig.lab.pos	= "top.left",fig.lab.size = 14)


smry <- pf.summary(PF.mult, taxadf,  factor=17)
p1 <- ggboxplot(x = "CO2", y= "Meanratio", data = data.frame(CO2 = CO2, Meanratio = smry$ilr),ylab='ILR balance', fill = "blue", add = "jitter") +
  theme(plot.margin=unit(c(1, .5, 0.5, 0.5),"cm"))
#ggboxplot(x = "CO2", y= "Meanratio", data = data.frame(CO2 = CO2, Meanratio = smry$MeanRatio), main='Ratio of Geometric Means',ylab='Group1/Group2')
p1.1 <- ggboxplot(x = "days", y= "Meanratio", data = data.frame(days = days, Meanratio = smry$ilr),ylab='ILR balance', fill = "blue", add = "jitter")+
  theme(plot.margin=unit(c(1, .5, 0.5, 0.5),"cm"))
#ggboxplot(x = "days", y= "Meanratio", data = data.frame(days = days, Meanratio = smry$MeanRatio), main='Ratio of Geometric Means',ylab='Group1/Group2')
p1.2 <- ggarrange(p1.1, p1) 
p1.3 <- annotate_figure(p1.2, fig.lab  = "o__Actinomycetales; f__Propionibacteriaceae" , fig.lab.pos	= "top.left",fig.lab.size = 14)

#smry <- pf.summary(PF.mult, taxadf,  factor=23)
#p1 <- ggboxplot(x = "CO2", y= "Meanratio", data = data.frame(CO2 = CO2, Meanratio = smry$ilr),ylab='ILR balance', fill = "magenta2", add = "jitter") +
#  theme(plot.margin=unit(c(1, .5, 0.5, 0.5),"cm"))
#ggboxplot(x = "CO2", y= "Meanratio", data = data.frame(CO2 = CO2, Meanratio = smry$MeanRatio), main='Ratio of Geometric Means',ylab='Group1/Group2')
#p1.1 <- ggboxplot(x = "days", y= "Meanratio", data = data.frame(days = days, Meanratio = smry$ilr),ylab='ILR balance', fill = "magenta2", add = "jitter")+
#  theme(plot.margin=unit(c(1, .5, 0.5, 0.5),"cm"))
#ggboxplot(x = "days", y= "Meanratio", data = data.frame(days = days, Meanratio = smry$MeanRatio), main='Ratio of Geometric Means',ylab='Group1/Group2')
#p1.2 <- ggarrange(p1.1, p1) 
#p1.4 <- annotate_figure(p1.2, fig.lab  = "f__Paenibacillaceae; g__Paenibacillus; s__" , fig.lab.pos	= "top.left",fig.lab.size = 14)


smry <- pf.summary(PF.mult, taxadf,  factor=24)
p1 <- ggboxplot(x = "CO2", y= "Meanratio", data = data.frame(CO2 = CO2, Meanratio = smry$ilr),ylab='ILR balance', fill = "deeppink3", add = "jitter") +
  theme(plot.margin=unit(c(1, .5, 0.5, 0.5),"cm"))
#ggboxplot(x = "CO2", y= "Meanratio", data = data.frame(CO2 = CO2, Meanratio = smry$MeanRatio), main='Ratio of Geometric Means',ylab='Group1/Group2')
p1.1 <- ggboxplot(x = "days", y= "Meanratio", data = data.frame(days = days, Meanratio = smry$ilr),ylab='ILR balance', fill = "deeppink3", add = "jitter")+
  theme(plot.margin=unit(c(1, .5, 0.5, 0.5),"cm"))
#ggboxplot(x = "days", y= "Meanratio", data = data.frame(days = days, Meanratio = smry$MeanRatio), main='Ratio of Geometric Means',ylab='Group1/Group2')
p1.2 <- ggarrange(p1.1, p1) 
p2.3 <- annotate_figure(p1.2, fig.lab  = "f__Paenibacillaceae; g__Cohnella" , fig.lab.pos	= "top.left",fig.lab.size = 14)

smry <- pf.summary(PF.mult, taxadf,  factor=25)
p1 <- ggboxplot(x = "CO2", y= "Meanratio", data = data.frame(CO2 = CO2, Meanratio = smry$ilr),ylab='ILR balance', fill = "firebrick1", add = "jitter" ) +
  theme(plot.margin=unit(c(1, .5, 0.5, 0.5),"cm"))
#ggboxplot(x = "CO2", y= "Meanratio", data = data.frame(CO2 = CO2, Meanratio = smry$MeanRatio), main='Ratio of Geometric Means',ylab='Group1/Group2')
p1.1 <- ggboxplot(x = "days", y= "Meanratio", data = data.frame(days = days, Meanratio = smry$ilr),ylab='ILR balance', fill = "firebrick1", add = "jitter")+
  theme(plot.margin=unit(c(1, .5, 0.5, 0.5),"cm"))
#ggboxplot(x = "days", y= "Meanratio", data = data.frame(days = days, Meanratio = smry$MeanRatio), main='Ratio of Geometric Means',ylab='Group1/Group2')
p1.2 <- ggarrange(p1.1, p1) 
p3.3 <- annotate_figure(p1.2, fig.lab  = "f__Paenibacillaceae" , fig.lab.pos	= "top.left",fig.lab.size = 14)


pdf(NULL)
boxplots <- ggarrange(p00.3, p0.3, p1.3, p2.3, p3.3, nrow = 6) 
x = dev.off()
boxplots
```

```{r phylofactortablesCHROM, eval=TRUE, echo=FALSE, message=FALSE, warning=FALSE, cache=TRUE}
#Tables to show taxa which form part of important bins 
taxtable <- tax_table(physeqtest)@.Data %>% as.data.frame %>% rownames_to_column("OTU_ID")

# sig. factors for CO2 1*, 3. , 6*, 11*, 12***, 14*, 17*, 21., 22*, 24***, 25**

bin1 <-  data.frame(OTU_ID = matrix(unlist(bin.projection$otus[["Bin 1"]]), nrow=length(bin.projection$otus[["Bin 1"]]), byrow=T),stringsAsFactors=FALSE)
bin1  <- bin1 %>% left_join(taxtable) %>%  dplyr::select(-OTU_ID, - Species, -Genus, -Family, -Class, -Order) %>% unique
bin1  <- bin1 %>% mutate(Bin = "Bin1") %>% dplyr::select(Bin, everything())

bin2 <-  data.frame(OTU_ID = matrix(unlist(bin.projection$otus[["Bin 2"]]), nrow=length(bin.projection$otus[["Bin 2"]]), byrow=T),stringsAsFactors=FALSE)
bin2  <- bin2 %>% left_join(taxtable) %>%  dplyr::select(-OTU_ID, - Species, - Kingdom) %>% unique
bin2  <- bin2 %>% mutate(Bin = "Bin2") %>% dplyr::select(Bin, everything())

bin3 <-  data.frame(OTU_ID = matrix(unlist(bin.projection$otus[["Bin 3"]]), nrow=length(bin.projection$otus[["Bin 3"]]), byrow=T),stringsAsFactors=FALSE)
bin3  <- bin3 %>% left_join(taxtable) %>%  dplyr::select(-OTU_ID, - Species, - Kingdom) %>% unique
bin3  <- bin3 %>% mutate(Bin = "Bin3") %>% dplyr::select(Bin, everything())

bin4 <-  data.frame(OTU_ID = matrix(unlist(bin.projection$otus[["Bin 4"]]), nrow=length(bin.projection$otus[["Bin 4"]]), byrow=T),stringsAsFactors=FALSE)
bin4  <- bin4 %>% left_join(taxtable) %>%  dplyr::select(-OTU_ID, - Species, - Kingdom) %>% unique
bin4  <- bin4 %>% mutate(Bin = "Bin4") %>% dplyr::select(Bin, everything())

bin5 <-  data.frame(OTU_ID = matrix(unlist(bin.projection$otus[["Bin 5"]]), nrow=length(bin.projection$otus[["Bin 5"]]), byrow=T),stringsAsFactors=FALSE)
bin5  <- bin5 %>% left_join(taxtable) %>%  dplyr::select(-OTU_ID, - Species, - Kingdom) %>% unique
bin5  <- bin5 %>% mutate(Bin = "Bin5") %>% dplyr::select(Bin, everything())

bin6 <-  data.frame(OTU_ID = matrix(unlist(bin.projection$otus[["Bin 6"]]), nrow=length(bin.projection$otus[["Bin 6"]]), byrow=T),stringsAsFactors=FALSE)
bin6  <- bin6 %>% left_join(taxtable) %>%  dplyr::select(-OTU_ID, - Species, - Kingdom) %>% unique
bin6  <- bin6 %>% mutate(Bin = "Bin6") %>% dplyr::select(Bin, everything())

bin7 <-  data.frame(OTU_ID = matrix(unlist(bin.projection$otus[["Bin 7"]]), nrow=length(bin.projection$otus[["Bin 7"]]), byrow=T),stringsAsFactors=FALSE)
bin7  <- bin7 %>% left_join(taxtable) %>%  dplyr::select(-OTU_ID, - Species, - Kingdom) %>% unique
bin7  <- bin7 %>% mutate(Bin = "Bin7") %>% dplyr::select(Bin, everything())

bin8 <-  data.frame(OTU_ID = matrix(unlist(bin.projection$otus[["Bin 8"]]), nrow=length(bin.projection$otus[["Bin 8"]]), byrow=T),stringsAsFactors=FALSE)
bin8  <- bin8 %>% left_join(taxtable) %>%  dplyr::select(-OTU_ID, - Species, - Kingdom) %>% unique
bin8  <- bin8 %>% mutate(Bin = "Bin8") %>% dplyr::select(Bin, everything())

bin9 <-  data.frame(OTU_ID = matrix(unlist(bin.projection$otus[["Bin 9"]]), nrow=length(bin.projection$otus[["Bin 9"]]), byrow=T),stringsAsFactors=FALSE)
bin9  <- bin9 %>% left_join(taxtable) %>%  dplyr::select(-OTU_ID, - Species, - Kingdom) %>% unique
bin9  <- bin9 %>% mutate(Bin = "Bin9") %>% dplyr::select(Bin, everything())

bin10 <-  data.frame(OTU_ID = matrix(unlist(bin.projection$otus[["Bin 10"]]), nrow=length(bin.projection$otus[["Bin 10"]]), byrow=T),stringsAsFactors=FALSE)
bin10  <- bin10 %>% left_join(taxtable) %>%  dplyr::select(-OTU_ID, - Species, - Kingdom) %>% unique
bin10  <- bin10 %>% mutate(Bin = "Bin10") %>% dplyr::select(Bin, everything())

bin11 <-  data.frame(OTU_ID = matrix(unlist(bin.projection$otus[["Bin 11"]]), nrow=length(bin.projection$otus[["Bin 11"]]), byrow=T),stringsAsFactors=FALSE)
bin11  <- bin11 %>% left_join(taxtable) %>%  dplyr::select(-OTU_ID, - Species, - Kingdom) %>% unique
bin11  <- bin11 %>% mutate(Bin = "Bin11") %>% dplyr::select(Bin, everything())

bin12 <-  data.frame(OTU_ID = matrix(unlist(bin.projection$otus[["Bin 12"]]), nrow=length(bin.projection$otus[["Bin 12"]]), byrow=T),stringsAsFactors=FALSE)
bin12  <- bin12 %>% left_join(taxtable) %>%  dplyr::select(-OTU_ID, - Species, - Kingdom) %>% unique
bin12  <- bin12 %>% mutate(Bin = "Bin12") %>% dplyr::select(Bin, everything())

bin13 <-  data.frame(OTU_ID = matrix(unlist(bin.projection$otus[["Bin 13"]]), nrow=length(bin.projection$otus[["Bin 13"]]), byrow=T),stringsAsFactors=FALSE)
bin13  <- bin13 %>% left_join(taxtable) %>%  dplyr::select(-OTU_ID, - Species, - Kingdom) %>% unique
bin13  <- bin13 %>% mutate(Bin = "Bin13") %>% dplyr::select(Bin, everything())

bin15 <-  data.frame(OTU_ID = matrix(unlist(bin.projection$otus[["Bin 15"]]), nrow=length(bin.projection$otus[["Bin 15"]]), byrow=T),stringsAsFactors=FALSE)
bin15  <- bin15 %>% left_join(taxtable) %>%  dplyr::select(-OTU_ID, - Species, - Kingdom) %>% unique
bin15  <- bin15 %>% mutate(Bin = "Bin15") %>% dplyr::select(Bin, everything())

bin18 <-  data.frame(OTU_ID = matrix(unlist(bin.projection$otus[["Bin 18"]]), nrow=length(bin.projection$otus[["Bin 18"]]), byrow=T),stringsAsFactors=FALSE)
bin18  <- bin18 %>% left_join(taxtable) %>%  dplyr::select(-OTU_ID, - Species, - Kingdom) %>% unique
bin18  <- bin18 %>% mutate(Bin = "Bin18") %>% dplyr::select(Bin, everything())

bin23 <-  data.frame(OTU_ID = matrix(unlist(bin.projection$otus[["Bin 23"]]), nrow=length(bin.projection$otus[["Bin 23"]]), byrow=T),stringsAsFactors=FALSE)
bin23  <- bin23 %>% left_join(taxtable) %>%  dplyr::select(-OTU_ID, - Species, - Kingdom) %>% unique
bin23  <- bin23 %>% mutate(Bin = "Bin23") %>% dplyr::select(Bin, everything())

bin24 <-  data.frame(OTU_ID = matrix(unlist(bin.projection$otus[["Bin 24"]]), nrow=length(bin.projection$otus[["Bin 24"]]), byrow=T),stringsAsFactors=FALSE)
bin24  <- bin24 %>% left_join(taxtable) %>%  dplyr::select(-OTU_ID, - Species, - Kingdom) %>% unique
bin24  <- bin24 %>% mutate(Bin = "Bin24") %>% dplyr::select(Bin, everything())

bin25 <-  data.frame(OTU_ID = matrix(unlist(bin.projection$otus[["Bin 25"]]), nrow=length(bin.projection$otus[["Bin 25"]]), byrow=T),stringsAsFactors=FALSE)
bin25  <- bin25 %>% left_join(taxtable) %>%  dplyr::select(-OTU_ID, - Species, - Kingdom) %>% unique
bin25  <- bin25 %>% mutate(Bin = "Bin25") %>% dplyr::select(Bin, everything())

bin26 <-  data.frame(OTU_ID = matrix(unlist(bin.projection$otus[["Bin 26"]]), nrow=length(bin.projection$otus[["Bin 26"]]), byrow=T),stringsAsFactors=FALSE)
bin26  <- bin26 %>% left_join(taxtable) %>%  dplyr::select(-OTU_ID, - Species, - Kingdom) %>% unique
bin26  <- bin26 %>% mutate(Bin = "Bin26") %>% dplyr::select(Bin, everything())

# sig. factors for CO2 1., 3.(interaction.), 4*(interaction*), 6(interation only**), 7*(interaction***), 8*, 9**(interaction***), 10(interaction only**), 13*(interaction.), 16*, 17***(interaction***),22*(interaction.), 24***(interaction*), 25**(interaction**)

bin <- rbind(bin7, bin8, bin9, bin10, bin18, bin23, bin24, bin25 , bin26)
kable(bin, booktabs=TRUE, caption = "Taxa represented in bins") %>% kable_styling(latex_options="scale_down")

```

go [back to top](#metadata)  

<br/>


# Bacillus, Paenibacillus, Symbiobacterium
```{r BacPae,  eval=TRUE, echo=FALSE, fig.width=8, fig.asp = 0.7, message=FALSE, warning=FALSE, cache=TRUE, fig.cap="Relative abundances of *Bacillus sp*. and *Paenibacillus sp*.. The abundances of these two genera were found to be significantly different in both soils at eleveated CO~2~ using Deseq2 and Aldex2 tests for abundances in day 14 or day 42"}

physeqB.flt <- prune_taxa(taxa_sums(physeqB) >= 10, physeqB)
tax_table(physeqB.flt)[, "Genus"] <- str_replace_all(tax_table(physeqB.flt)[, "Genus"], "D_5__", "")
physeqtest <- tax_glom(physeqB.flt, "Genus")
barplot.rel   <- phyloseq::transform_sample_counts(physeqtest , function(x) x / sum(x) )
subs <- subset_taxa(barplot.rel, Genus %in% c("Bacillus", "Paenibacillus", "Symbiobacterium"))
#subs <- subset_taxa(barplot.rel, Species %in% c(" s__horti"))
subs <- psmelt(subs)
barplot.relsoil1 <- subs   %>% ggbarplot(data = ., x = "Genus", y = "Abundance", add = "mean_se", facet.by = c("Soils","days"), fill = "CO2", width = 0.5, xlab = "", position = position_dodge(0.7), scales = "free") + theme(axis.text.x = element_text(angle = 45, hjust = 1))
barplot.relsoil1 
```

go [back to top](#metadata)  

<br/>

# Alpha diversity {#alphadiversity}
## by soil
```{r alphasoil, warning=FALSE, message=FALSE, echo=FALSE, cache=TRUE, fig.cap = "alpha diversity"}

as.data.frame(sample_data(physeqB.flt)) %>% melt(., measure.var = c("Observed", "InvSimpson", "Shannon", "pielou_ev"))  %>%
ggboxplot(., x = "Soils", y = "value", add = "jitter", facet.by = "variable", scales = "free", ylab = "alpha diversity") +
  stat_compare_means( method = "wilcox") 
```
  
go [back to top](#metadata) 


## by CO2
```{r alphaCO2, echo=FALSE, fig.height=10,fig.asp= 0.9, message=FALSE, warning=FALSE, fig.cap = "alpha diversity"}
p1 <- as.data.frame(sample_data(physeqB.flt)) %>% melt(., measure.var = c("Observed"))  %>%
ggboxplot(., x = "CO2", y = "value", add = "jitter", facet.by = c("Soils","variable"), scales = "free", ylab = "alpha diversity") + stat_compare_means( method = "wilcox") 
p2 <- as.data.frame(sample_data(physeqB.flt)) %>% melt(., measure.var = c("InvSimpson"))  %>%
ggboxplot(., x = "CO2", y = "value", add = "jitter", facet.by = c("Soils","variable"), scales = "free", ylab = "alpha diversity") + stat_compare_means( method = "wilcox") 
p3 <- as.data.frame(sample_data(physeqB.flt)) %>% melt(., measure.var = c("Shannon"))  %>%
ggboxplot(., x = "CO2", y = "value", add = "jitter", facet.by = c("Soils","variable"), scales = "free", ylab = "alpha diversity") + stat_compare_means( method = "wilcox") 
p4 <- as.data.frame(sample_data(physeqB.flt)) %>% melt(., measure.var = c("pielou_ev"))  %>%
ggboxplot(., x = "CO2", y = "value", add = "jitter", facet.by = c("Soils","variable"), scales = "free", ylab = "alpha diversity") + stat_compare_means( method = "wilcox") 

pdf(NULL)
gg1 <- ggarrange(p1,p2,p3,p4)
x = dev.off()
gg1

```
  
go [back to top](#metadata) 

## by days and CO2
```{r alphadaysV, warning=FALSE, message=FALSE, echo=FALSE, fig.height=10,fig.asp= 0.9, fig.cap = "alpha diversity"}

p1 <- as.data.frame(sample_data(physeqB.flt)) %>% melt(., measure.var = c("Observed"))  %>% filter(Soils == "Vertisol") %>% 
ggboxplot(., x = "days", y = "value", add = "jitter", facet.by = c("CO2","variable"), scales = "free", ylab = "alpha diversity") + stat_compare_means( method = "kruskal") 
p2 <- as.data.frame(sample_data(physeqB.flt)) %>% melt(., measure.var = c("InvSimpson"))   %>% filter(Soils == "Vertisol") %>% 
ggboxplot(., x = "days", y = "value", add = "jitter", facet.by = c("CO2","variable"), scales = "free", ylab = "alpha diversity") + stat_compare_means( method = "kruskal") 
p3 <- as.data.frame(sample_data(physeqB.flt)) %>% melt(., measure.var = c("Shannon"))   %>% filter(Soils == "Vertisol") %>% 
ggboxplot(., x = "days", y = "value", add = "jitter", facet.by = c("CO2","variable"), scales = "free", ylab = "alpha diversity") + stat_compare_means( method = "kruskal") 
p4 <- as.data.frame(sample_data(physeqB.flt)) %>% melt(., measure.var = c("pielou_ev"))   %>% filter(Soils == "Vertisol") %>% 
ggboxplot(., x = "days", y = "value", add = "jitter", facet.by = c("CO2","variable"), scales = "free", ylab = "alpha diversity") + stat_compare_means( method = "kruskal") 
pdf(NULL)
gg1 <- ggarrange(p1,p2,p3,p4)
gg1 <- annotate_figure(gg1, fig.lab  = "Vertisol",fig.lab.pos	= "top.left",fig.lab.size = 14)
x = dev.off()
gg1
```
  
go [back to top](#metadata) 

```{r alphadaysC, echo=FALSE, message=FALSE, warning=FALSE,fig.height=10,fig.asp= 0.9, fig.cap = "alpha diversity"}

p1 <- as.data.frame(sample_data(physeqB.flt)) %>% melt(., measure.var = c("Observed"))  %>% filter(Soils == "Chromosol") %>% 
ggboxplot(., x = "days", y = "value", add = "jitter", facet.by = c("CO2","variable"), scales = "free", ylab = "alpha diversity") + stat_compare_means( method = "kruskal") 
p2 <- as.data.frame(sample_data(physeqB.flt)) %>% melt(., measure.var = c("InvSimpson"))   %>% filter(Soils == "Chromosol") %>% 
ggboxplot(., x = "days", y = "value", add = "jitter", facet.by = c("CO2","variable"), scales = "free", ylab = "alpha diversity") + stat_compare_means( method = "kruskal") 
p3 <- as.data.frame(sample_data(physeqB.flt)) %>% melt(., measure.var = c("Shannon"))   %>% filter(Soils == "Chromosol") %>% 
ggboxplot(., x = "days", y = "value", add = "jitter", facet.by = c("CO2","variable"), scales = "free", ylab = "alpha diversity") + stat_compare_means( method = "kruskal") 
p4 <- as.data.frame(sample_data(physeqB.flt)) %>% melt(., measure.var = c("pielou_ev"))   %>% filter(Soils == "Chromosol") %>% 
ggboxplot(., x = "days", y = "value", add = "jitter", facet.by = c("CO2","variable"), scales = "free", ylab = "alpha diversity") + stat_compare_means( method = "kruskal") 
pdf(NULL)
gg1 <- ggarrange(p1,p2,p3,p4)
gg1 <- annotate_figure(gg1, fig.lab  = "Chromosol",fig.lab.pos	= "top.left",fig.lab.size = 14)
x = dev.off()
gg1
```
  
<br/>

(Fig. \@ref(fig:alphadaysC))  
The alpha diversity in the Chromosol was more sensitive to elevated CO~2~ compared to the Vertisol. Over time the number of ASVs reduced and this was most significant in the Chromosol and at elevated CO~2~. In the Chromosol, lower ASV numbers after 3 days occured along with greater diversity (Shannon, InvSimpson and Pilou Evenness) but exclusively at elevated CO~2~ concentrations. At ambient CO~2~ no changes in alpha diversity were observed in this soil.   
  
In the Vertisol diversity increased after 3 days under both, ambient and elevated CO~2~ concentrations but after 14 days the diversity under elevated CO~2~ maintained a higher diversity compared to ambient CO~2~. 

<br/>

go [back to top](#metadata) 

  

<br/>
# Beta diversity 
## Permanova  {#perm}
```{r permanova, warning=FALSE, message=FALSE, echo=FALSE}
ps.perm <- rarefy_even_depth(physeqB.flt, sample.size = min(colSums(otu_table(physeqB.flt))),  rngseed = TRUE, replace = TRUE, trimOTUs = TRUE)
ps.perm <- phyloseq::transform_sample_counts(ps.perm, function(x) x / sum(x) )
otutable <- as.data.frame(phyloseq::otu_table(ps.perm))
otutable <- as.data.frame(t(otutable))
meta.df <- data.frame(sample_data(ps.perm))

require(vegan)
perm.soil <- adonis(otutable ~ Soils*CO2, data = meta.df ,method = "bray") 

perma.df <- data.frame(perm.soil$aov.tab[1:3,5:6])

kable(perma.df , booktabs=TRUE, digits = 2)
#perma.df  <- ggtexttable(format(perma.df, digits = 1))
#perma.df <- annotate_figure(perma.df , top =  "PERMANOVAs (filtered and rarefied ASVs, bray curtis)")
#perma.df
```

  
go [back to top](#metadata) 

## NMDS Bray
```{r nmds, message=FALSE, warning=FALSE, echo=FALSE, results=FALSE, cache = TRUE, fig.cap = "beta diversity"}
ps.perm <- rarefy_even_depth(physeqB.flt, sample.size = min(colSums(otu_table(physeqB.flt))),  rngseed = TRUE, replace = TRUE, trimOTUs = TRUE)
ps.perm <- phyloseq::transform_sample_counts(ps.perm, function(x) x / sum(x) )
ord_nmds <- phyloseq::ordinate(ps.perm, "NMDS", "bray")

nmds1 <- phyloseq::plot_ordination(ps.perm, ord_nmds , color = "Soils", shape = "CO2") + geom_point(aes(size = InvSimpson)) + ggtitle("nMDS, bray") + labs(subtitle ="filtered and rarefied OTU table")
nmds1  <- nmds1  +  geom_text_repel(label = nmds1$data$days)
nmds1 <- nmds1 + stat_ellipse(aes(group = days), type = "t", linetype = 2)
#nmds1 <- nmds1 + geom_polygon(aes(alpha = 0.6, fill = Rhizosphere,group = TreatmentID), stat = "identity") + guides(alpha = F)
nmds1 

```
  
go [back to top](#metadata) 
  
# Correlations of metadata
```{r corr1, warning=FALSE, message=FALSE, echo=FALSE, cache = TRUE}
require(psych)
require(reshape2)
test.df <- data.frame(sample_data(physeqB))
correlation_matrix=corr.test(test.df[,c("phytate","Olsen.P", "Microbial.P", "microbial.C", "Microbial.C.P","Cumulative.C","respiration.rate","DOC","Shannon","Simpson","InvSimpson","Fisher","pielou_ev","Observed")], method="spearman", adjust="holm")

r_matrix=correlation_matrix$r
r_matrix[row(r_matrix)>col(r_matrix)]=0
diag(r_matrix)=0
p_matrix=correlation_matrix$p
p_matrix[row(p_matrix)>col(p_matrix)]=1
diag(p_matrix)=1
r_table=melt(r_matrix)
p_table=melt(p_matrix)
cortable <- data.frame(r_table$Var1, r_table$Var2, r_table$value, p_table$value) #creates the table with r and p values
```
## Significant correlations accross all samples (<0.05 BH corrected) {#cor_meta}
```{r corr2, warning=FALSE, message=FALSE, echo=FALSE, cache = TRUE}
#filter1
cortable.sig <- cortable %>% 
  filter(p_table.value <= 0.05) %>%
  arrange(p_table.value)
cortable.sig

test.df <- data.frame(sample_data(physeqB))
df <- test.df[,c("phytate","Olsen.P", "Microbial.P", "microbial.C", "Microbial.C.P","Cumulative.C","respiration.rate","DOC","Shannon","Simpson","InvSimpson","Fisher","pielou_ev","Observed")] 
ggcorrplot::ggcorrplot(cor(df))
```

# Abundance Heatmaps
## Bacteria
### Chromosol, taxa filtered to those present in 25% of samples  
Correlation threshold = p.adj < 0.15
```{r heatmapch_bac, eval=TRUE, include=TRUE,  message= FALSE, warning=FALSE, echo= FALSE, cache = FALSE}
#https://microbiome.github.io/tutorials/Composition.html

### https://microbiome.github.io/tutorials/  Cross correaltion heatmap
# Define data sets to cross-correlate
physeq_nc.ch <-  phyloseq::prune_samples(sample_data(physeqB)$Soil == "Chromosol", physeqB)
physeq_nc.ch <-  phyloseq::tax_glom(physeq_nc.ch, "Genus")
physeq_nc.ch.flt.clr = phyloseq::filter_taxa(physeq_nc.ch, function(x) sum(x > 0) > (0.25*length(x)), TRUE)
## clr transform
physeq_nc.ch.flt.clr <- microbiome::transform(physeq_nc.ch.flt.clr, "clr")
otutable <- as.data.frame(phyloseq::otu_table(physeq_nc.ch.flt.clr)) 
otutable <- as.data.frame(t(otutable))
taxtable_nc.ch <-  data.frame(phyloseq::tax_table(physeq_nc.ch)@.Data) %>% rownames_to_column("Id")
# Get sample metadata
meta.df <- as.data.frame(sample_data(physeq_nc.ch.flt.clr))

x <- as.matrix(otutable)
y <- data.frame(meta.df[,c("phytate","Olsen.P", "Microbial.P", "microbial.C", "Microbial.C.P","Cumulative.C","respiration.rate")])

# ...Or, alternatively, the same output is also available in a handy table format
correlation.table <- microbiome::associate(x, y, method = "spearman", mode = "table", p.adj.threshold = 0.15, n.signif = 1) %>% dplyr::rename(Id = X1) %>%  dplyr::left_join(taxtable_nc.ch)

p <- microbiome::heat(correlation.table, "Genus", "X2", fill = "Correlation", star = "p.adj", p.adj.threshold = 0.15)

#results <- p$data %>% dplyr::rename(Feature.ID = Id ) %>% inner_join(taxtableb) %>% dplyr::select( Phylum, Family, Genus, p.adj) %>% filter(p.adj < 0.05) %>% dplyr::select(Phylum, Family, Genus, p.adj) %>% data.frame() 

#results <- ggtexttable(format(results, digits = 1),rows = NULL) 

#fig <- ggarrange(p,results, ncol = 2, nrow = 1, widths = c(0.4,0.6)) + ggtitle( "Correlations of clr transformed abundances on genus level")
#fig <- annotate_figure(fig , top =  "CHROMOSOL, Correlations of clr transformed abundances on genus level")

#pdf(NULL)
#x = dev.off()
#fig
```
  
```{r, heatmapch_bac_plot, fig.height=8, fig.width=12.5, message= FALSE, warning=FALSE, echo= FALSE, cache = FALSE}
p + ggtitle( "Correlations of clr transformed abundances on genus level")
```

### Vertisol, taxa filtered to those present in 25% of samples  
Correlation threshold = p.adj < 0.05
```{r heatmapvert_bac, eval=TRUE, include=TRUE,  message= FALSE, warning=FALSE, echo= FALSE, cache = FALSE}
#https://microbiome.github.io/tutorials/Composition.html

### https://microbiome.github.io/tutorials/  Cross correaltion heatmap
# Define data sets to cross-correlate
physeq_nc.ch <-  prune_samples(sample_data(physeqB)$Soil == "Vertisol", physeqB)
physeq_nc.ch <-  tax_glom(physeq_nc.ch, "Genus")
physeq_nc.ch.flt.clr = filter_taxa(physeq_nc.ch, function(x) sum(x > 0) > (0.25*length(x)), TRUE)
## clr transform
physeq_nc.ch.flt.clr <- microbiome::transform(physeq_nc.ch.flt.clr, "clr")
otutable <- as.data.frame(phyloseq::otu_table(physeq_nc.ch.flt.clr))
otutable <- as.data.frame(t(otutable))
taxtable_nc.ch <-  data.frame(phyloseq::tax_table(physeq_nc.ch)@.Data) %>% rownames_to_column("Id")
# Get sample metadata
meta.df <- as.data.frame(sample_data(physeq_nc.ch.flt.clr))

x <- as.matrix(otutable)
y <- as.data.frame(meta.df[,c("phytate","Olsen.P", "Microbial.P", "microbial.C", "Microbial.C.P","Cumulative.C","respiration.rate")])

# ...Or, alternatively, the same output is also available in a handy table format
correlation.table <- microbiome::associate(x, y, method = "spearman", mode = "table", p.adj.threshold = 0.05, n.signif = 1) %>% dplyr::rename(Id = X1) %>%  dplyr::left_join(taxtable_nc.ch)

p <- microbiome::heat(correlation.table, "Genus", "X2", fill = "Correlation", star = "p.adj", p.adj.threshold = 0.05)


#results <- p$data %>% rename(Feature.ID = Id ) %>% inner_join(taxtableb) %>% dplyr::select( Phylum, Family, Genus, p.adj) %>% filter(p.adj < 0.1) %>% dplyr::select(Phylum, Family, Genus, p.adj) %>% data.frame() 

#results <- ggtexttable(format(results, digits = 1),rows = NULL) 
```

```{r, heatmapch_bacvert_plot, fig.height=8, fig.width=12.5, message= FALSE, warning=FALSE, echo= FALSE, cache = FALSE}
p + ggtitle( "Correlations of clr transformed abundances on genus level")
```
